<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<link rel="alternate" hreflang="en" href="https://openbiox.github.io/Bizard/Correlation/Heatmap.html" />
<link rel="alternate" hreflang="zh" href="https://openbiox.github.io/Bizard/zh/Correlation/Heatmap.zh.html" />
<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="[Editor] YuJie Peng;">
<meta name="author" content="[Contributors] ShiXiang Wang, Peng Luo, Hong Yang, KeXin Li, Ying Shi.">

<title>Heatmap – A Biomedical Visualization Atlas</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../Correlation/Correlogram.html" rel="next">
<link href="../Correlation/Scatter.html" rel="prev">
<link href="../images/favicon.ico" rel="icon">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-d6a4bb07b738ea43132b6af4314c9077.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="../site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="../site_libs/d3v3-3.5.3/./d3v3.min.js"></script>
<link href="../site_libs/d3heatmapcore-0.0.0/heatmapcore.css" rel="stylesheet">
<script src="../site_libs/d3heatmapcore-0.0.0/heatmapcore.js"></script>
<script src="../site_libs/d3v3-tip-0.6.6/index.js"></script>
<script src="../site_libs/d3heatmap-binding-0.9.0/d3heatmap.js"></script>
<script src="../site_libs/plotly-binding-4.11.0/plotly.js"></script>
<script src="../site_libs/typedarray-0.1/typedarray.min.js"></script>
<script src="../site_libs/jquery-3.5.1/jquery.min.js"></script>
<link href="../site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="../site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="../site_libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="../site_libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


</head>

<body class="nav-sidebar docked quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../Correlation/Scatter.html">Correlation</a></li><li class="breadcrumb-item"><a href="../Correlation/Heatmap.html">Heatmap</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-center sidebar-header">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../images/Bizard_logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/openbiox/Bizard" title="Bizard GitHub website" class="quarto-navigation-tool px-1" aria-label="Bizard GitHub website"><i class="bi bi-github"></i></a>
    <a href="https://x.com/PengL_Robin" title="Peng Lab Twitter" class="quarto-navigation-tool px-1" aria-label="Peng Lab Twitter"><i class="bi bi-twitter"></i></a>
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-translate"></i></a>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../GraphGallery.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Graph Gallery</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">𝐁𝐀𝐒𝐈𝐂𝐒</span></span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Distribution</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Distribution/ViolinPlot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Violin Plot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Distribution/Density.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Density Plot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Distribution/Histogram.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Histogram</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Distribution/BoxPlot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Box Plot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Distribution/Ridgeline.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Ridgeline Plot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Distribution/Beeswarm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Beeswarm Plot</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Correlation</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Correlation/Scatter.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scatter Plot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Correlation/Heatmap.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Heatmap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Correlation/Correlogram.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlogram</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Correlation/Bubble.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bubble Plot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Correlation/ConnectedScatter.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Connected Scatter</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Correlation/Density2D.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2D Density</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Ranking</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Ranking/BarPlot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bar Plot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Ranking/Radar.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Radar/Spider Chart</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Ranking/Wordcloud.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Wordcloud</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Ranking/Parallel.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Parallel Coordinates Plot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Ranking/Lollipop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lollipop Plot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Ranking/CircularBarplot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Circular Barplot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Ranking/Table.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Table</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Composition/Part Of The Whole</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Composition/GroupedBarplot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Grouped and Stacked Barplot</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Composition/Treemap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Treemap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Composition/Donut.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Donut Chart</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Composition/PieChart.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Pie Chart</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Composition/Dendrogram.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Dendrogram</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Composition/CircularPacking.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Circular Packing Chart</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Composition/Waffle.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Waffle Chart</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false">
 <span class="menu-text">Data Over Time/Evolution</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataOverTime/LineChart.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Line Chart</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataOverTime/AreaChart.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Area Chart</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataOverTime/StackedArea.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Stacked Area Chart</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataOverTime/Streamgraph.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Streamgraph</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../DataOverTime/Timeseries.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Timeseries</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" role="navigation" aria-expanded="false">
 <span class="menu-text">Proportion/Flow</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Proportion/ChordDiagram.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chord Diagram</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Proportion/Network.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Network Graph</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Proportion/Sankey.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Sankey Diagram</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Proportion/ArcDiagram.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Arc Diagram</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Proportion/EdgeBundling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Hierarchical Edge Bundling</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="false">
 <span class="menu-text">Animation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Animation/Animation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Animation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../Animation/Interactivity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interactivity</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">𝐎𝐌𝐈𝐂𝐒</span></span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">𝐂𝐋𝐈𝐍𝐈𝐂𝐒</span></span>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../About.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About Us</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#example" id="toc-example" class="nav-link active" data-scroll-target="#example">Example</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup">Setup</a></li>
  <li><a href="#data-preparation" id="toc-data-preparation" class="nav-link" data-scroll-target="#data-preparation">Data Preparation</a></li>
  <li><a href="#visualization" id="toc-visualization" class="nav-link" data-scroll-target="#visualization">Visualization</a>
  <ul class="collapse">
  <li><a href="#basic-r-heatmaps" id="toc-basic-r-heatmaps" class="nav-link" data-scroll-target="#basic-r-heatmaps">1. Basic R Heatmaps</a></li>
  <li><a href="#ggplot2-heatmaps" id="toc-ggplot2-heatmaps" class="nav-link" data-scroll-target="#ggplot2-heatmaps">2. ggplot2 Heatmaps</a></li>
  <li><a href="#lattice-heatmaps" id="toc-lattice-heatmaps" class="nav-link" data-scroll-target="#lattice-heatmaps">3. lattice Heatmaps</a></li>
  <li><a href="#pheatmap-heatmaps" id="toc-pheatmap-heatmaps" class="nav-link" data-scroll-target="#pheatmap-heatmaps">4. pheatmap Heatmaps</a></li>
  <li><a href="#complexheatmap-heatmaps" id="toc-complexheatmap-heatmaps" class="nav-link" data-scroll-target="#complexheatmap-heatmaps">5. complexheatmap Heatmaps</a></li>
  </ul></li>
  <li><a href="#applications" id="toc-applications" class="nav-link" data-scroll-target="#applications">Applications</a>
  <ul class="collapse">
  <li><a href="#basic-heatmap-4" id="toc-basic-heatmap-4" class="nav-link" data-scroll-target="#basic-heatmap-4">1. Basic Heatmap</a></li>
  <li><a href="#heatmaps-with-color-annotations-1" id="toc-heatmaps-with-color-annotations-1" class="nav-link" data-scroll-target="#heatmaps-with-color-annotations-1">2. Heatmaps with Color Annotations</a></li>
  <li><a href="#comparative-heatmaps" id="toc-comparative-heatmaps" class="nav-link" data-scroll-target="#comparative-heatmaps">3. Comparative Heatmaps</a></li>
  </ul></li>
  <li><a href="#reference" id="toc-reference" class="nav-link" data-scroll-target="#reference">Reference</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/openbiox/Bizard/blob/main/Correlation/Heatmap.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/openbiox/Bizard/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../Correlation/Scatter.html">Correlation</a></li><li class="breadcrumb-item"><a href="../Correlation/Heatmap.html">Heatmap</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Heatmap</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Authors</div>
    <div class="quarto-title-meta-contents">
             <p><strong>[Editor]</strong> YuJie Peng; </p>
             <p><strong>[Contributors]</strong> ShiXiang Wang, Peng Luo, Hong Yang, KeXin Li, Ying Shi. </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<p>A heatmap is a powerful visualization tool that represents matrix values through color gradients. It is widely used to illustrate gene expression differences across sample groups, variations in compound concentrations, and pairwise sample similarities. More broadly, any tabular dataset can be structured into a heatmap to enhance interpretability.</p>
<section id="example" class="level2">
<h2 class="anchored" data-anchor-id="example">Example</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/Heatmap_demo.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:60.0%" alt="Heatmap Example"></p>
</figure>
</div>
<p>In this heatmap, rows correspond to car models (e.g., <code>Mazda RX4</code>, <code>Merc 450SLC</code>), while columns represent specific attributes (e.g., <code>mpg</code> for miles per gallon, <code>hp</code> for horsepower, <code>wt</code> for weight). Each cell encodes the attribute value for a given car, with higher values appearing closer to yellow.</p>
<p>Hierarchical clustering (dendrograms) on the top and left depict similarity relationships among rows and columns, facilitating pattern recognition and comparative analysis.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>Notes</strong>: The heatmap in this example provides limited insight due to the disparity in attribute scales. Attributes such as hp (horsepower) and disp (displacement) exhibit substantially higher values than others, distorting the color mapping and overshadowing lower-range features. To enhance interpretability, data normalization is required, as discussed in the following sections.</p>
</div>
</div>
</section>
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
<ul>
<li><p>System Requirements: Cross-platform (Linux/MacOS/Windows)</p></li>
<li><p>Programming Language: R</p></li>
<li><p>Dependencies:<code>readr</code>, <code>ggplot2</code>, <code>hrbrthemes</code>, <code>dplyr</code>, <code>tidyr</code>, <code>viridis</code>, <code>tibble</code>, <code>htmlwidgets</code>, <code>RColorBrewer</code>, <code>plotly</code>, <code>devtools</code>, <code>d3heatmap</code>, <code>heatmaply</code>, <code>lattice</code>, <code>ComplexHeatmap</code>, <code>pheatmap</code>, <code>circlize</code>, <code>gridExtra</code>, <code>cowplot</code></p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Installing necessary packages</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"readr"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"readr"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"tidyverse"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"ggplot2"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"hrbrthemes"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"hrbrthemes"</span>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"dplyr"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"tidyr"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"tidyr"</span>)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"viridis"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"viridis"</span>)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"tibble"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"tibble"</span>)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"htmlwidgets"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"htmlwidgets"</span>)</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"RColorBrewer"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"RColorBrewer"</span>)</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"plotly"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"plotly"</span>)</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"devtools"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"devtools"</span>)</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"d3heatmap"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>  devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"talgalili/d3heatmap"</span>)</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"heatmaply"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"heatmaply"</span>)</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"lattice"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"lattice"</span>)</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"ComplexHeatmap"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>  devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"jokergoo/ComplexHeatmap"</span>)</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"pheatmap"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"pheatmap"</span>)</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"circlize"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"circlize"</span>)</span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"gridExtra"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"gridExtra"</span>)</span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"cowplot"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"cowplot"</span>)</span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Loading the libraries</span></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hrbrthemes)     <span class="co"># Additional themes for ggplot2</span></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)          <span class="co"># Data manipulation functions</span></span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)          <span class="co"># Data tidying functions</span></span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)        <span class="co"># Color palettes for better visualization</span></span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)         <span class="co"># Tidy data frames</span></span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(htmlwidgets)    <span class="co"># Interactive web visualizations</span></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)   <span class="co"># Color palettes for visualizations</span></span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)         <span class="co"># Interactive plots</span></span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(d3heatmap)      <span class="co"># D3.js-based interactive heatmaps</span></span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(heatmaply)      <span class="co"># Interactive heatmaps with ggplot2</span></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lattice)        <span class="co"># Trellis graphics for multivariate data</span></span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap) <span class="co"># Advanced heatmap functionalities</span></span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)       <span class="co"># Pretty heatmaps with clustering support</span></span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(circlize)       <span class="co"># Circular visualizations</span></span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra)      <span class="co"># Grid layout visualization</span></span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)        <span class="co"># Data visualization package</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-preparation" class="level2">
<h2 class="anchored" data-anchor-id="data-preparation">Data Preparation</h2>
<p>This section provides a brief tutorial using built-in R datasets (<code>mtcars</code>, <code>volcano</code>) and the <code>TCGA-CHOL.methylation450.tsv</code> dataset from <a href="https://xenabrowser.net/datapages/?hub=https://gdcv18.xenahubs.net:443">UCSC Xena DATASETS</a>. These datasets will be used to demonstrate heatmap visualization in R. This example demonstrates how to load and work with these datasets in R.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load built-in R datasets `mtcars`</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">"mtcars"</span>, <span class="at">package =</span> <span class="st">"datasets"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>mtcars_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(mtcars)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Load and process methylation data</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>raw_methylation_data <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(<span class="st">"https://bizard-1301043367.cos.ap-guangzhou.myqcloud.com/TCGA-CHOL.methylation450.tsv"</span>)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to matrix and clean up row/column names</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>methylation_matrix <span class="ot">&lt;-</span> raw_methylation_data[, <span class="sc">-</span><span class="dv">1</span>] <span class="sc">%&gt;%</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(raw_methylation_data<span class="sc">$</span>Composite)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Tidy to long format</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>methylation_long <span class="ot">&lt;-</span> methylation_matrix <span class="sc">%&gt;%</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"Composite"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>Composite, <span class="at">names_to =</span> <span class="st">"Sample"</span>, <span class="at">values_to =</span> <span class="st">"Methylation_Level"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">Methylation_Level =</span> <span class="fu">as.numeric</span>(Methylation_Level),</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">Composite =</span> <span class="fu">gsub</span>(<span class="st">"^cg0+"</span>, <span class="st">"cg"</span>, Composite),</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sample =</span> <span class="fu">substr</span>(Sample, <span class="dv">9</span>, <span class="dv">12</span>)</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Standardize methylation values</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>methylation_long_standardized <span class="ot">&lt;-</span> methylation_long <span class="sc">%&gt;%</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Composite) <span class="sc">%&gt;%</span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Standardized_Level =</span> <span class="fu">scale</span>(Methylation_Level)[,<span class="dv">1</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert back to wide format matrix</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>standardized_methylation_matrix <span class="ot">&lt;-</span> methylation_long_standardized <span class="sc">%&gt;%</span></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Composite, Sample, Standardized_Level) <span class="sc">%&gt;%</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Sample, <span class="at">values_from =</span> Standardized_Level) <span class="sc">%&gt;%</span></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"Composite"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up raw methylation matrix (numerical version for raw heatmap)</span></span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>methylation_matrix_num <span class="ot">&lt;-</span> methylation_matrix <span class="sc">%&gt;%</span></span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(.)))) <span class="sc">%&gt;%</span></span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(<span class="fu">gsub</span>(<span class="st">"^cg0+"</span>, <span class="st">"cg"</span>, <span class="fu">rownames</span>(.))) <span class="sc">%&gt;%</span></span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>  { <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(., <span class="fu">substr</span>(<span class="fu">colnames</span>(.), <span class="dv">9</span>, <span class="dv">12</span>)) } <span class="sc">%&gt;%</span></span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="visualization" class="level2">
<h2 class="anchored" data-anchor-id="visualization">Visualization</h2>
<section id="basic-r-heatmaps" class="level3">
<h3 class="anchored" data-anchor-id="basic-r-heatmaps">1. Basic R Heatmaps</h3>
<section id="basic-heatmap" class="level4">
<h4 class="anchored" data-anchor-id="basic-heatmap">1.1 Basic Heatmap</h4>
<p>Using the <code>heatmap()</code> function to construct a heatmap. Below is the simplest heatmap created in base R without additional parameters. The <code>heatmap()</code> function requires a matrix as input. If the data is in a data frame, it can be converted into a matrix using the <code>as.matrix()</code> function. A matrix can also be transposed using the <code>t()</code> function, which swaps rows and columns, effectively exchanging the X and Y axes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(mtcars_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-BasicHeatBasicR" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-BasicHeatBasicR-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-BasicHeatBasicR-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-BasicHeatBasicR-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Basic Heatmap Basic R
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap uses color variations to display numerical differences between performance variables of various cars in the <code>mtcars</code> dataset.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<p>This heatmap is not particularly insightful because the variables <code>hp</code> and <code>disp</code> with relatively high values dominate the scale. Consequently, variations in other variables (e.g., <code>mpg</code>, <code>wt</code>) are hard to discern. Standardizing the data, as discussed in the next section, is necessary.</p>
</div>
</div>
</section>
<section id="standardization" class="level4">
<h4 class="anchored" data-anchor-id="standardization">1.2 Standardization</h4>
<p>The <code>scale</code> parameter in the <code>heatmap()</code> function can be used to standardize the matrix. It can be applied to rows or columns. Here, columns are chosen since we aim to capture variations between columns.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(mtcars_matrix, <span class="at">scale=</span><span class="st">"column"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-StandardizationBasicR" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-StandardizationBasicR-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-StandardizationBasicR-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-StandardizationBasicR-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Standardization Basic R
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap presents standardized values of performance variables in the <code>mtcars</code> dataset, making comparisons across variables on a common scale and improving the visual accuracy and interpretability.</p>
</section>
<section id="clustering-dendrograms-and-reordering" class="level4">
<h4 class="anchored" data-anchor-id="clustering-dendrograms-and-reordering">1.3 Clustering Dendrograms and Reordering</h4>
<p>In the generated heatmap, the order of rows and columns differs from the original <code>mtcars</code> matrix because <code>heatmap()</code> applies clustering algorithms to reorder variables and observations. It calculates distances between pairs of rows and columns and sorts them based on similarities.</p>
<p>Additionally, dendrograms are displayed alongside the heatmap. These can be disabled using the <code>Rowv</code> and <code>Colv</code> parameters, as shown below.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(mtcars_matrix, <span class="at">Colv =</span> <span class="cn">NA</span>, <span class="at">Rowv =</span> <span class="cn">NA</span>, <span class="at">scale=</span><span class="st">"column"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-ClusteringandReorderingBasicR" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ClusteringandReorderingBasicR-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-ClusteringandReorderingBasicR-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ClusteringandReorderingBasicR-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Clustering Dendrograms and Reordering Basic R
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap shows the standardized values of performance variables in the <code>mtcars</code> dataset, with the dendrograms removed to maintain the original order of rows and columns.</p>
</section>
<section id="color-palettes" class="level4">
<h4 class="anchored" data-anchor-id="color-palettes">1.4 Color Palettes</h4>
<p>Heatmaps can utilize different color palettes:</p>
<ul>
<li>Built-in R palettes: <code>terrain.colors()</code>, <code>rainbow()</code>, <code>heat.colors()</code>, <code>topo.colors()</code>, <code>cm.colors()</code></li>
<li>Palettes suggested by the RColorBrewer package</li>
<li>Custom color palettes</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Native color palette</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(mtcars_matrix, <span class="at">scale=</span><span class="st">"column"</span>, <span class="at">col =</span> <span class="fu">cm.colors</span>(<span class="dv">256</span>))</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(mtcars_matrix, <span class="at">scale=</span><span class="st">"column"</span>, <span class="at">col =</span> <span class="fu">terrain.colors</span>(<span class="dv">256</span>))</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Using Rcolorbrewer</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>coul <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"PiYG"</span>))(<span class="dv">25</span>)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(mtcars_matrix, <span class="at">scale=</span><span class="st">"column"</span>, <span class="at">col =</span> coul)</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Custom color palette</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>my_palette <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"white"</span>, <span class="st">"red"</span>))(<span class="dv">256</span>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(mtcars_matrix, <span class="at">scale =</span> <span class="st">"column"</span>, <span class="at">col =</span> my_palette)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-ColorPalettesBasicR-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ColorPalettesBasicR-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-ColorPalettesBasicR-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ColorPalettesBasicR-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Color Palettes Basic R
</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display">
<div id="fig-ColorPalettesBasicR-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ColorPalettesBasicR-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-ColorPalettesBasicR-2.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ColorPalettesBasicR-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Color Palettes Basic R
</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display">
<div id="fig-ColorPalettesBasicR-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ColorPalettesBasicR-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-ColorPalettesBasicR-3.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ColorPalettesBasicR-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Color Palettes Basic R
</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display">
<div id="fig-ColorPalettesBasicR-4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ColorPalettesBasicR-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-ColorPalettesBasicR-4.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ColorPalettesBasicR-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: Color Palettes Basic R
</figcaption>
</figure>
</div>
</div>
</div>
<p>The following four heatmaps demonstrate the same data using different color palettes. These highlight the differences in performance variables of cars in the <code>mtcars</code> dataset, emphasizing how palette selection can influence visual interpretation and clarity.</p>
<section id="rcolorbrewer" class="level5">
<h5 class="anchored" data-anchor-id="rcolorbrewer"><strong>RColorBrewer</strong></h5>
<p>The <code>display.brewer.all()</code> function in the <code>RColorBrewer</code> package showcases all available color schemes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">plt=</span><span class="fu">c</span>(<span class="fl">0.1</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">1</span>))</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">display.brewer.all</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-RColorBrewerBasicR" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-RColorBrewerBasicR-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-RColorBrewerBasicR-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-RColorBrewerBasicR-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8: RColorBrewer Basic R
</figcaption>
</figure>
</div>
</div>
</div>
<p>The image displays color scheme names on the left and their respective colors on the right. These schemes fall into three categories: - <em>Sequential (seq)</em>: Gradual variations of a single color, suitable for continuous or ordered categorical variables. - <em>Qualitative (qual)</em>: Highly distinct colors without hierarchical relationships, ideal for unordered categorical variables. - <em>Diverging (div)</em>: Two colors transitioning in opposite directions, often used for continuous variables with meaningful midpoint values (e.g., temperature or odds ratios).</p>
</section>
</section>
<section id="customizing-layout" class="level4">
<h4 class="anchored" data-anchor-id="customizing-layout">1.5 Customizing Layout</h4>
<p>Heatmaps can be customized using standard parameters like <code>main</code> for the title and <code>xlab</code>/<code>ylab</code> for axis labels. Additionally, <code>labRow</code>/<code>colRow</code> allow changes to labels, while <code>cexRow</code>/<code>cexCol</code> adjust their sizes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Custom titles and axis titles</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> <span class="fu">heatmap</span>(mtcars_matrix, <span class="at">Colv =</span> <span class="cn">NA</span>, <span class="at">Rowv =</span> <span class="cn">NA</span>, <span class="at">scale=</span><span class="st">"column"</span>, <span class="at">col =</span> coul, <span class="at">xlab=</span><span class="st">"variable"</span>, <span class="at">main=</span><span class="st">"heatmap"</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Custom labels</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>p6 <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">4</span>, <span class="dv">2</span>))  <span class="co"># Adjust margins</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(mtcars_matrix, <span class="at">scale=</span><span class="st">"column"</span>, <span class="at">cexRow=</span>.<span class="dv">8</span>, <span class="at">labRow=</span><span class="fu">paste</span>(<span class="st">"new_"</span>, <span class="fu">rownames</span>(mtcars_matrix),<span class="at">sep=</span><span class="st">""</span>), <span class="at">col=</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Blues"</span>))(<span class="dv">25</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-CustomizingLayoutBasicR-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-CustomizingLayoutBasicR-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-CustomizingLayoutBasicR-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-CustomizingLayoutBasicR-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;9: Customizing Layout Basic R
</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display">
<div id="fig-CustomizingLayoutBasicR-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-CustomizingLayoutBasicR-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-CustomizingLayoutBasicR-2.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-CustomizingLayoutBasicR-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10: Customizing Layout Basic R
</figcaption>
</figure>
</div>
</div>
</div>
<p>The following two heatmaps illustrate standardized performance variables in the <code>mtcars</code> dataset with customized titles, axis labels, and row annotations, improving visual clarity and data communication.</p>
</section>
<section id="heatmaps-with-color-annotations" class="level4">
<h4 class="anchored" data-anchor-id="heatmaps-with-color-annotations">1.6 Heatmaps with Color Annotations</h4>
<p>Heatmaps are often used to compare observed structures against expected ones. The<code>RowSideColors</code> parameter can add a color vector to the side of a heatmap to indicate expected groupings.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>my_group <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.factor</span>(<span class="fu">substr</span>(<span class="fu">rownames</span>(mtcars_matrix), <span class="dv">1</span> , <span class="dv">1</span>))) <span class="co"># Group row labels by the first letter and convert to numeric</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>colSide <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[my_group] <span class="co"># Use the Set1 color scheme from RColorBrewer to generate 9 colors, assigning each group a color</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>colMain <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Blues"</span>))(<span class="dv">25</span>)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(mtcars_matrix, <span class="at">Colv =</span> <span class="cn">NA</span>, <span class="at">Rowv =</span> <span class="cn">NA</span>, <span class="at">scale=</span><span class="st">"column"</span> , <span class="at">RowSideColors=</span>colSide, <span class="at">col=</span>colMain)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-ColorAnnotationsBasicR" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ColorAnnotationsBasicR-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-ColorAnnotationsBasicR-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ColorAnnotationsBasicR-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;11: Heatmaps with Color Annotations Basic R
</figcaption>
</figure>
</div>
</div>
</div>
<p>The heatmap below groups rows by their first letters and assigns a distinct color to each group, helping observers quickly identify differences between groups and understand data patterns more intuitively.</p>
</section>
</section>
<section id="ggplot2-heatmaps" class="level3">
<h3 class="anchored" data-anchor-id="ggplot2-heatmaps">2. ggplot2 Heatmaps</h3>
<p>The <code>ggplot2</code> package uses the <code>geom_tile()</code> function to create heatmaps. Unlike <code>heatmap()</code>, it doesn’t natively support standardization, clustering, or dendrograms.</p>
<section id="basic-heatmap-1" class="level4">
<h4 class="anchored" data-anchor-id="basic-heatmap-1">2.1 Basic Heatmap</h4>
<p>A basic heatmap created with <code>ggplot2</code> and <code>geom_tile()</code>. Standardization must be performed beforehand, as <code>geom_tile()</code> cannot directly standardize data.</p>
<p>The input data must be in a long-format data frame, where each row represents one observation with three key variables:</p>
<ul>
<li><code>x</code>: Position on the X-axis</li>
<li><code>y</code>: Position on the Y-axis</li>
<li><code>fill</code>: Numerical values mapped to colors</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(methylation_long_standardized, <span class="fu">aes</span>(<span class="at">x =</span> Sample, <span class="at">y =</span> Composite, <span class="at">fill=</span> Standardized_Level)) <span class="sc">+</span> </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>()</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>p </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>This heatmap demonstrates the distribution of standardized methylation levels (Standardized_Level) in the TCGA-CHOL dataset across samples (Sample) and composites (Composite), highlighting differences in methylation patterns.</p>
</section>
<section id="color-palettes-1" class="level4">
<h4 class="anchored" data-anchor-id="color-palettes-1">2.2 Color Palettes</h4>
<p>Heatmaps can have their color palettes customized, just like any ggplot2 chart. Below are three examples using different approaches:</p>
<ul>
<li><code>scale_fill_gradient()</code>: Sets the extreme colors of the palette.</li>
<li><code>scale_fill_distiller())</code>: Uses <code>ColorBrewer</code> palettes.</li>
<li><code>scale_fill_viridis()</code> Implements the <code>Viridis</code> palette. Remember to set <code>discrete=FALSE</code> for continuous variables (ensuring smooth color gradients).</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extreme colors</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>p7 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(methylation_long_standardized, <span class="fu">aes</span>(<span class="at">x =</span> Sample, <span class="at">y =</span> Composite, <span class="at">fill=</span> Standardized_Level)) <span class="sc">+</span> </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>()<span class="sc">+</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low=</span><span class="st">"white"</span>, <span class="at">high=</span><span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_ipsum</span>() <span class="sc">+</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>))</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co"># ColorBrewer color palette</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>p8 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(methylation_long_standardized, <span class="fu">aes</span>(<span class="at">x =</span> Sample, <span class="at">y =</span> Composite, <span class="at">fill=</span> Standardized_Level)) <span class="sc">+</span> </span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_distiller</span>(<span class="at">palette =</span> <span class="st">"RdPu"</span>) <span class="sc">+</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_ipsum</span>() <span class="sc">+</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) </span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Using Viridis</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>p9 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(methylation_long_standardized, <span class="fu">aes</span>(<span class="at">x =</span> Sample, <span class="at">y =</span> Composite, <span class="at">fill=</span> Standardized_Level)) <span class="sc">+</span> </span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>()<span class="sc">+</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">discrete=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_ipsum</span>() <span class="sc">+</span> </span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>))</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(p7, p8, p9, <span class="at">ncol =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>These three heatmaps showcase identical data from the TCGA-CHOL dataset, illustrating the distribution of standardized methylation levels between various samples and composites. They emphasize the importance of palette choice for visual impact and data interpretation.</p>
</section>
<section id="handling-wide-format-data" class="level4">
<h4 class="anchored" data-anchor-id="handling-wide-format-data">2.3 Handling Wide-Format Data</h4>
<p>A common issue involves input in a wide matrix format, such as the <code>volcano</code> dataset. In such cases, the <code>gather()</code> function from the <code>tidyr</code> package can transform the data into a long format suitable for visualization with ggplot.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>volcano <span class="sc">%&gt;%</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowid_to_column</span>(<span class="at">var=</span><span class="st">"X"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(<span class="at">key=</span><span class="st">"Y"</span>, <span class="at">value=</span><span class="st">"Z"</span>, <span class="sc">-</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Y=</span><span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">"V"</span>,<span class="st">""</span>,Y))) <span class="sc">%&gt;%</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(X, Y, <span class="at">fill=</span> Z)) <span class="sc">+</span> </span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_ipsum</span>() <span class="sc">+</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>This heatmap represents the elevation values (Z) in the <code>volcano</code> dataset. The x-axis indicates row indices (X), the y-axis represents column indices (Y), and color fill depicts changes in elevation.</p>
</section>
<section id="interactive-heatmaps" class="level4">
<h4 class="anchored" data-anchor-id="interactive-heatmaps">2.4 Interactive Heatmaps</h4>
<p>One of the significant advantages of creating heatmaps with <code>ggplot2</code> is the ability to make them interactive using the <code>plotly</code> package. Simply wrap the plot in an object and pass it to the <code>ggplotly()</code> function.</p>
<p>You can also customize tooltips to provide richer information. Try hovering over a cell to view tooltips or zoom in on specific areas.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create tooltips text</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>methylation_long_annotated <span class="ot">&lt;-</span> methylation_long_standardized <span class="sc">%&gt;%</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">text =</span> <span class="fu">paste0</span>(<span class="st">"Composite: "</span>,Composite, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="st">"Sample: "</span>, Sample, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="st">"Standardized_Level: "</span>, Standardized_Level))</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(methylation_long_annotated, <span class="fu">aes</span>(Composite, Sample, <span class="at">fill =</span> Standardized_Level, <span class="at">text =</span> text)) <span class="sc">+</span> </span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_ipsum</span>()</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>pp <span class="ot">&lt;-</span> <span class="fu">ggplotly</span>(p, <span class="at">tooltip=</span><span class="st">"text"</span>, <span class="at">width =</span> <span class="dv">1200</span>, <span class="at">height =</span> <span class="dv">700</span>)</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>pp</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>This interactive heatmap, built with the <code>plotly</code> package, allows dynamic exploration and in-depth data analysis, offering a more intuitive understanding and information retrieval process.</p>
<p>Alternative packages like <code>d3heatmap</code> and <code>heatmaply</code> can also create interactive heatmaps.</p>
<section id="using-the-d3heatmap-package" class="level5">
<h5 class="anchored" data-anchor-id="using-the-d3heatmap-package"><strong>Using the <code>d3heatmap</code> Package</strong></h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">d3heatmap</span>(standardized_methylation_matrix, <span class="at">scale =</span> <span class="st">"row"</span>, <span class="at">dendrogram =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-d3heatmapggplot2" class="cell-output-display quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-d3heatmapggplot2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="d3heatmap html-widget html-fill-item" id="htmlwidget-1fe2f2622ae4da613ef2" style="width:95%;height:480px;"></div>
<script type="application/json" data-for="htmlwidget-1fe2f2622ae4da613ef2">{"x":{"rows":null,"cols":null,"matrix":{"data":["-0.98","-0.58","0.984","1.824","0.586","-0.106","-0.954","-0.539","-0.701","1.295","-0.828","-1.153","0.802","-1.075","0.647","1.725","0.071","-0.21","0.529","-0.932","0.847","-1.251","-0.162","-2.051","-0.155","1.356","1.34","0.292","0.091","-1.356","0.196","0.359","0.089","0.427","-1.747","1.745","1.285","0.113","-0.574","-0.548","-0.086","-1.048","-0.123","0.556","0.545","0.67","-0.263","0.53","0.93","0.238","0.492","-0.765","-0.256","0.496","-2.616","-0.938","-0.623","-0.361","1.634","-0.59","0.306","-1.285","1.14","0.923","0.809","-1.015","-1.031","-0.592","0.505","-0.851","1.503","0.379","0.67","-0.731","1.716","-0.637","-0.932","0.016","2.727","-0.732","-0.553","-0.673","-0.352","-0.335","0.602","0.15","-0.047","-0.802","0.626","-2.059","0.094","-0.338","0.143","0.547","-1.512","0.895","1.303","0.371","-0.071","-0.581","0.983","0.296","-0.613","2.473","-0.137","-0.334","0.122","-0.72","-0.329","-1.159","0.699","-2.73","0.276","0.628","0.744","0.697","0.33","-0.319","0.046","0.179","-0.55","-0.303","1.906","-0.767","-0.211","-0.474","-0.717","-0.729","1.614","0.964","-0.556","-0.727","-0.743","1.777","-0.067","-1.24","-0.152","-0.294","-0.107","1.952","-0.721","0.125","-0.531","0.5","1.13","-1.362","-1.04","0.71","-0.051","-0.58","-0.016","1.678","0.354","-1.323","0.257","0.943","-0.964","-2.431","0.311","-0.302","-0.007","0.488","1.3","0.265","0.14","0.366","0.099","0.443","0.677","0.36","-0.676","-0.273","0.49","0.659","0.596","-2.741","-2.579","0.616","-1.206","0.441","0.588","0.114","0.457","0.41","0.659","0.356","0.145","0.312","2.825","-0.701","-0.375","-0.518","-0.819","-0.269","0.155","0.094","-0.348","-0.353","0.167","0.662","-1.592","0.629","1.136","-0.022","-0.663","0.173","1.083","0.318","-1.89","-1.403","0.113","-0.464","0.443","0.881","-0.72","1.58","0.857","0.87","-1.105","-1.051"],"x":[-0.98,-0.58,0.984,1.824,0.586,-0.106,-0.954,-0.539,-0.701,1.295,-0.828,-1.153,0.802,-1.075,0.647,1.725,0.07099999999999999,-0.21,0.529,-0.9320000000000001,0.847,-1.251,-0.162,-2.051,-0.155,1.356,1.34,0.292,0.091,-1.356,0.196,0.359,0.089,0.427,-1.747,1.745,1.285,0.113,-0.574,-0.548,-0.08599999999999999,-1.048,-0.123,0.556,0.545,0.67,-0.263,0.53,0.93,0.238,0.492,-0.765,-0.256,0.496,-2.616,-0.9379999999999999,-0.623,-0.361,1.634,-0.59,0.306,-1.285,1.14,0.923,0.8090000000000001,-1.015,-1.031,-0.592,0.505,-0.851,1.503,0.379,0.67,-0.731,1.716,-0.637,-0.9320000000000001,0.016,2.727,-0.732,-0.553,-0.673,-0.352,-0.335,0.602,0.15,-0.047,-0.802,0.626,-2.059,0.094,-0.338,0.143,0.547,-1.512,0.895,1.303,0.371,-0.07099999999999999,-0.581,0.983,0.296,-0.613,2.473,-0.137,-0.334,0.122,-0.72,-0.329,-1.159,0.699,-2.73,0.276,0.628,0.744,0.697,0.33,-0.319,0.046,0.179,-0.55,-0.303,1.906,-0.767,-0.211,-0.474,-0.717,-0.729,1.614,0.964,-0.556,-0.727,-0.743,1.777,-0.067,-1.24,-0.152,-0.294,-0.107,1.952,-0.721,0.125,-0.531,0.5,1.13,-1.362,-1.04,0.71,-0.051,-0.58,-0.016,1.678,0.354,-1.323,0.257,0.9429999999999999,-0.964,-2.431,0.311,-0.302,-0.007,0.488,1.3,0.265,0.14,0.366,0.099,0.443,0.677,0.36,-0.676,-0.273,0.49,0.659,0.596,-2.741,-2.579,0.616,-1.206,0.441,0.588,0.114,0.457,0.41,0.659,0.356,0.145,0.312,2.825,-0.701,-0.375,-0.518,-0.819,-0.269,0.155,0.094,-0.348,-0.353,0.167,0.662,-1.592,0.629,1.136,-0.022,-0.663,0.173,1.083,0.318,-1.89,-1.403,0.113,-0.464,0.443,0.881,-0.72,1.58,0.857,0.87,-1.105,-1.051],"dim":[20,11],"rows":["cg29","cg165","cg236","cg289","cg292","cg321","cg363","cg622","cg658","cg714","cg721","cg734","cg769","cg905","cg924","cg948","cg957","cg1245","cg1249","cg1261"],"cols":["A8I3","AA2U","A8Y6","AA38","AA30","A95A","AA36","AA2G","A8Y2","AAVB","AA2Q"]},"title":null,"image":"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAALCAYAAAB/Ca1DAAACTElEQVQokVVTT0gUURj/7RIxCSFSemhJjYXtVmukBAqSHmzp3kZkkqcuwhxWiU7dSrosEuF0EAo1YVkGukTB7hBZ4FLEDkRl2b5Z2CjFFRlnGhp5vw7LvOqDB9/7vt/73vf7/kBaOqXIs1Rv8tKdMoUbkJ7JyC4tnUbFobR0hkaWpXqToZGlcAPmilX6oU16JoUbUFo6EQHomXyTPEl6Jo2KQ6Pi0A9t+qHNvalRpfuhTeEGdMbSzBWrCidFnlLkCXqmCijcQJ3oXqo3VRBp6cq+hNRfFiLPXLFKKfKM/zqYBAA8/HAGGzs+NnZ8JJYnsL+4gu7teSQ72qCtLuD+i88IhibhyAy6397E+2IBONEL1ARinT0oza62dCnyRE20nADqR26gS1vHod8bYOUl6mfv4nv6NNKfCtAaZQSJEWwGKRy9dRFr0wV8/eFivG+3hd9yELfi1xAMTQIARm4fQ0/8GbRGGVHm3dvz6J9JYzNI4UnvA0SyNl1AsqMN4327Ch8kRloZxjp7kJ1rx8q5p4gNDINbDlATAICZ5nVk+o/jvHz8n+3ehW/YX1zBgauX1SfZuXZAWrrqknADGhWHe1OjajTomZQiT3qmGpPQyNIPbZbqTUpL5xJSanxi9Ew6MoPnH39ivG8X/8pmkAIAdGnr0FYXEBsYhiMzyv/68CkAwKBrI7E8gcaVR4gJN2CXtq4eRwEiGRx8hXezX1TTUBOIDQyrGmuNMgCoZmFvalRRjTYkouOHNp2xtBrsqCwR1VyxylK9qbYnNLL8Awvc5xZrlI5qAAAAAElFTkSuQmCC","rowcolors":null,"colcolors":null,"theme":null,"options":{"hm_width":null,"hm_height":null,"xaxis_height":80,"yaxis_width":120,"xaxis_font_size":13.92303081346953,"yaxis_font_size":11.62346144208289,"xaxis_angle":60,"xaxis_location":"bottom","yaxis_location":"right","xaxis_title":null,"yaxis_title":null,"xaxis_title_font_size":14,"yaxis_title_font_size":14,"bins":16,"manual_breaks":false,"symbreaks":false,"print_values":false,"show_legend":false,"legend_scaler":1,"legend_title":null,"legend_location":"fl","legend_colors":["#A50026","#C62327","#E14730","#F46D43","#FB9957","#FEBF70","#FEE090","#FFF5AF","#F6FBD2","#E0F3F8","#BDE2EE","#99CAE1","#74ADD1","#5687BE","#4060AA","#313695"],"legend_bins":16,"legend_breaks":[-2.740793569864528,-2.392948623830973,-2.045103677797419,-1.697258731763864,-1.34941378573031,-1.001568839696755,-0.6537238936632006,-0.3058789476296462,0.04196599840390824,0.3898109444374627,0.7376558904710171,1.085500836504572,1.433345782538126,1.78119072857168,2.129035674605235,2.476880620638789,2.824725566672344],"legend_type":"histogram","legend_fill":null,"brush_color":"#0000FF","na_color":"#777777","show_grid":true,"cellnote_row":null,"cellnote_col":null,"cellnote_val":"Value","cellnote_fontsize":12,"cellnote_color":"#222222","anim_duration":500,"rsc_labs":null,"csc_labs":null,"rsc_cols":null,"csc_cols":null,"rsc_colnames":null,"csc_colnames":null},"params":{"x":[[-0.9797195170117419,-0.580362804852211,0.9837887820618177,1.824068329283986,0.5856988215700039,-0.1062494672209157,-0.9538755651595963,-0.5393862782373897,-0.7007406002103521,1.294788333969294,-0.8280100341928952],[-1.153315093651538,0.8015818480505704,-1.074510973830642,0.6466623173500413,1.724931396545329,0.07053634849203032,-0.2098537305115458,0.5288389836236399,-0.931536368107392,0.8474781062754468,-1.25081283423594],[-0.1622910638553522,-2.051117560849757,-0.1554301540722379,1.355921893280027,1.340324642025288,0.292343847212905,0.09132510219873061,-1.355773169530563,0.196223657776918,0.3592026274091246,0.0892701784049038],[0.4273430564271093,-1.746639732243097,1.744794413609897,1.284721353379774,0.1126961756186584,-0.5742888537828471,-0.5475196023829294,-0.08623911275640996,-1.047808385457544,-0.1229901170988138,0.5559308046862019],[0.5446250247615636,0.6698576863046591,-0.263187304437315,0.5297583368887064,0.9297673527283611,0.2382085388111039,0.4923173879451295,-0.7648929117710808,-0.2558134971217754,0.4956545870638722,-2.616295201173227],[-0.9380777703884358,-0.6232493802753217,-0.3608488937281428,1.634189089769844,-0.5899402448983968,0.3060841641390942,-1.285487757345285,1.140341611552798,0.9225072791536808,0.8094007790174252,-1.01491887699726],[-1.03123152341657,-0.5919687616868571,0.5052138902351452,-0.8511550251664654,1.503191457292569,0.3792607030976146,0.6701544980295049,-0.7306082982949109,1.7162300590541,-0.6368154004409478,-0.9322715987031841],[0.01648439846268823,2.727110138745493,-0.7321049760067055,-0.5531793552800508,-0.6731068900250576,-0.3524081504518815,-0.3354792057840342,0.6018785377056263,0.1502654509605802,-0.04712290886210393,-0.8023370394645556],[0.6255088331063493,-2.058508081020846,0.09433435100802916,-0.3383808676018802,0.1430076505238128,0.547297678876142,-1.512025395036625,0.8954327178912873,1.302718312258476,0.371233274149613,-0.07061847415435382],[-0.5812364502176446,0.9827641929415253,0.296222277238368,-0.61338531798845,2.473092442783439,-0.1370533759328337,-0.3338418109852037,0.1218383636732914,-0.7203351261575425,-0.3286815188700898,-1.159383676484858],[0.6993313710313994,-2.730012906323243,0.2761989024594144,0.6284064075363408,0.7441580021909731,0.6967828919474059,0.3299040413741468,-0.3189313805390221,0.04581544927489829,0.1787221915529295,-0.5503749705052274],[-0.3032909066346155,1.906492215644157,-0.7672704712988313,-0.2108216300356217,-0.4740247562376553,-0.7166361660106847,-0.729198818179753,1.613522507966276,0.9640052441214322,-0.5558466614836424,-0.7269305578510598],[-0.7425540862173636,1.777287466757848,-0.06685084000996684,-1.239706095445379,-0.1520756883434385,-0.2943416632781947,-0.1069169824937568,1.951927430983049,-0.7212612917013556,0.1250187867652998,-0.5305270370167449],[0.499798265119983,1.130436938627703,-1.361639985289646,-1.040364758286956,0.7102167523025055,-0.05118697958905728,-0.5795479722496718,-0.01601689119730373,1.678134572832591,0.3536286186899376,-1.323458560960085],[0.2571319380122459,0.9427348473315386,-0.9640986233551498,-2.430927708873259,0.3109929824191842,-0.3016547241493217,-0.006917948052824516,0.4880186183184128,1.300264581505766,0.2648216699512449,0.139634366892165],[0.3655827716513391,0.09889096215617685,0.4434679699428289,0.6768920275286943,0.3601059942708811,-0.6759430015239031,-0.2725594134011372,0.4895973702787476,0.6589574654609771,0.5958014234999237,-2.740793569864528],[-2.579346797934115,0.6156531615098081,-1.206485814298326,0.4406137936354844,0.5880888402819767,0.1136175369959042,0.4569123920016908,0.4102434281919565,0.6589202378838195,0.356479563895126,0.1453036578366743],[0.3117173115583144,2.824725566672344,-0.701061186704062,-0.3753966568702108,-0.5184565871630079,-0.8194407577296164,-0.2694639892102674,0.1546402071004765,0.09378671287145024,-0.3481174008932389,-0.3529332196321819],[0.1669186792454887,0.6620963207437908,-1.591637765410307,0.629000464126466,1.135595483177351,-0.02245257455841768,-0.662608203294802,0.1727517042448156,1.082811900910074,0.317623001610968,-1.890099010795425],[-1.403245621361144,0.1129382884951206,-0.4641982951429439,0.4434723303107238,0.8809449009079381,-0.7198464833397702,1.579517326031448,0.8566330332918525,0.8700164315680381,-1.105173962458927,-1.051057948302337]],"Rowv":true,"Colv":true,"distfun":["function (x, method = \"euclidean\", diag = FALSE, upper = FALSE, ","    p = 2) ","{","    if (!is.na(pmatch(method, \"euclidian\"))) ","        method <- \"euclidean\"","    METHODS <- c(\"euclidean\", \"maximum\", \"manhattan\", \"canberra\", ","        \"binary\", \"minkowski\")","    method <- pmatch(method, METHODS)","    if (is.na(method)) ","        stop(\"invalid distance method\")","    if (method == -1) ","        stop(\"ambiguous distance method\")","    x <- as.matrix(x)","    N <- nrow(x)","    attrs <- if (method == 6L) ","        list(Size = N, Labels = dimnames(x)[[1L]], Diag = diag, ","            Upper = upper, method = METHODS[method], p = p, call = match.call(), ","            class = \"dist\")","    else list(Size = N, Labels = dimnames(x)[[1L]], Diag = diag, ","        Upper = upper, method = METHODS[method], call = match.call(), ","        class = \"dist\")","    .Call(C_Cdist, x, method, attrs, p)","}"],"hclustfun":["function (d, method = \"complete\", members = NULL) ","{","    METHODS <- c(\"ward.D\", \"single\", \"complete\", \"average\", \"mcquitty\", ","        \"median\", \"centroid\", \"ward.D2\")","    if (method == \"ward\") {","        message(\"The \\\"ward\\\" method has been renamed to \\\"ward.D\\\"; note new \\\"ward.D2\\\"\")","        method <- \"ward.D\"","    }","    i.meth <- pmatch(method, METHODS)","    if (is.na(i.meth)) ","        stop(\"invalid clustering method\", paste(\"\", method))","    if (i.meth == -1) ","        stop(\"ambiguous clustering method\", paste(\"\", method))","    n <- as.integer(attr(d, \"Size\"))","    if (is.null(n)) ","        stop(\"invalid dissimilarities\")","    if (is.na(n) || n > 65536L) ","        stop(\"size cannot be NA nor exceed 65536\")","    if (n < 2) ","        stop(\"must have n >= 2 objects to cluster\")","    len <- as.integer(n * (n - 1)/2)","    if (length(d) != len) ","        (if (length(d) < len) ","            stop","        else warning)(\"dissimilarities of improper length\")","    if (is.null(members)) ","        members <- rep(1, n)","    else if (length(members) != n) ","        stop(\"invalid length of members\")","    storage.mode(d) <- \"double\"","    hcl <- .Fortran(C_hclust, n = n, len = len, method = as.integer(i.meth), ","        ia = integer(n), ib = integer(n), crit = double(n), members = as.double(members), ","        nn = integer(n), disnn = double(n), diss = d)","    hcass <- .Fortran(C_hcass2, n = n, ia = hcl$ia, ib = hcl$ib, ","        order = integer(n), iia = integer(n), iib = integer(n))","    structure(list(merge = cbind(hcass$iia[1L:(n - 1)], hcass$iib[1L:(n - ","        1)]), height = hcl$crit[1L:(n - 1)], order = hcass$order, ","        labels = attr(d, \"Labels\"), method = METHODS[i.meth], ","        call = match.call(), dist.method = attr(d, \"method\")), ","        class = \"hclust\")","}"],"dendrogram":"none","reorderfun":["function (d, w) ","reorder(d, w)"],"k_row":null,"k_col":null,"symm":false,"revC":null,"scale":"row","scale.by.range":false,"na.rm":true,"na.value":null,"digits":3,"cellnote":null,"cellnote_scale":false,"labRow":["cg29","cg165","cg236","cg289","cg292","cg321","cg363","cg622","cg658","cg714","cg721","cg734","cg769","cg905","cg924","cg948","cg957","cg1245","cg1249","cg1261"],"labCol":["A8I3","AA2U","A8Y6","AA38","AA30","A95A","AA36","AA2G","A8Y2","AAVB","AA2Q"],"col":"RdYlBu","symbreaks":false,"na.color":"#777777","rng":null,"breaks":null,"RowSideColors":null,"ColSideColors":null,"RowColorsPalette":["blue","orange","black"],"ColColorsPalette":["cyan","maroon","grey"]}},"evals":[],"jsHooks":[]}</script>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-d3heatmapggplot2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12: Using d3heatmap ggplot2
</figcaption>
</figure>
</div>
</div>
<p>The <code>d3heatmap</code> package creates interactive heatmaps, enabling dynamic exploration and deeper insights into data.</p>
</section>
<section id="using-the-heatmaply-package" class="level5">
<h5 class="anchored" data-anchor-id="using-the-heatmaply-package"><strong>Using the <code>heatmaply</code> Package</strong></h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">heatmaply</span>(standardized_methylation_matrix,</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">dendrogram =</span> <span class="st">"none"</span>,</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">xlab =</span> <span class="st">""</span>, <span class="at">ylab =</span> <span class="st">""</span>,</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">main =</span> <span class="st">""</span>,</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">scale =</span> <span class="st">"row"</span>,</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">50</span>,<span class="dv">0</span>,<span class="dv">30</span>,<span class="dv">30</span>),</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">grid_color =</span> <span class="st">"white"</span>,</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">grid_width =</span> <span class="fl">0.00001</span>,</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">titleX =</span> <span class="cn">FALSE</span>,</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">hide_colorbar =</span> <span class="cn">TRUE</span>,</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">branches_lwd =</span> <span class="fl">0.1</span>,</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">fontsize_row =</span> <span class="dv">5</span>, <span class="at">fontsize_col =</span> <span class="dv">5</span>,</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>               <span class="at">labCol =</span> <span class="fu">colnames</span>(standardized_methylation_matrix),</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">labRow =</span> <span class="fu">rownames</span>(standardized_methylation_matrix),</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>               <span class="at">heatmap_layers =</span> <span class="fu">theme</span>(<span class="at">axis.line =</span> <span class="fu">element_blank</span>())</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-heatmaplyggplot2" class="cell-output-display quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-heatmaplyggplot2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="plotly html-widget html-fill-item" id="htmlwidget-611bc1c929bbe50073f3" style="width:95%;height:480px;"></div>
<script type="application/json" data-for="htmlwidget-611bc1c929bbe50073f3">{"x":{"data":[{"x":[1,2,3,4,5,6,7,8,9,10,11],"y":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20],"z":[[0.24032761647023457,0.51275214195658569,0.40905353460669053,0.57214175749949037,0.65074584812695158,0.36311924134040191,0.77626377520359702,0.64637754626334265,0.64878224525868433,0.29388446383518507,0.30360790792530157],[0.52245121753715362,0.61142362592355703,0.2064777384215912,0.605477036610791,0.69650089379693514,0.48842541524304051,0.3734036871649084,0.52349928239080457,0.68701685808124435,0.54952943228555429,0.15285089103088489],[0.54846831113804384,1,0.36649454132138393,0.42500921386934232,0.39930452634906621,0.34522436541840146,0.44404295808279942,0.52024504919170567,0.50931102978828091,0.42991068942045263,0.4290453939069887],[0.02900839400057692,0.60307882320262318,0.27568097744803011,0.57162814203880963,0.5981261277663833,0.51287418780426308,0.57455663765015497,0.56617126286932651,0.61085295447637422,0.55651109227645612,0.518567478953467],[0.55814673623578637,0.5102281498555209,0.57214097403836328,0.6140820853452077,0.55716268115555068,0.3710077205170606,0.44348677920443591,0.58042940126397202,0.61085964344396981,0.59951190742659766,0],[0.53866053360517663,0.66184812716107755,0.31923256446027082,0.055676003152525667,0.54833816530234358,0.43825899900381132,0.4912166421033724,0.58014573465145958,0.72608826817992578,0.54004220739882458,0.51754883346767722],[0.58226227517761431,0.69557401807822961,0.24780322387554587,0.30552923633205914,0.62006979717518573,0.48326248177255754,0.38832776324971624,0.48958176439991691,0.79398310099905334,0.55599884083409346,0.25466357659249289],[0.3590391901680825,0.81179867066879563,0.48044803445207707,0.26971203181473496,0.4651350247861894,0.4395729933845185,0.47324903980290561,0.84317758788043784,0.36286503174613477,0.51492273880008088,0.39713573498250787],[0.43796501340333915,0.83501389025866235,0.35459820551326238,0.45457968569724,0.40728793810910557,0.36369606396023796,0.36143883478522787,0.78237375005061338,0.66566994436591953,0.39258636162743699,0.36184639071541996],[0.61811393627451139,0.00193704545376707,0.54208644302699416,0.60537029785461161,0.62616828485545384,0.61765603126664559,0.55173606197487735,0.43515476811970877,0.50069166070164373,0.52457204616389408,0.39356950279435754],[0.38802438131416817,0.66904051023046629,0.54568419811285929,0.38224794483410057,0.93681934869642591,0.46783420019859351,0.43247569540782183,0.51435128750973691,0.36303144309449081,0.43340288512517222,0.28414418396262969],[0.60484966817768382,0.12259152688283283,0.50940942818080159,0.43166012789195823,0.51815493750737096,0.59079686334286419,0.22078231063140308,0.65334898659937868,0.72652914902006205,0.55916200585568188,0.47977107439642769],[0.49542152325486821,0.98246067877369792,0.36091666286277885,0.39306561722572991,0.37151730667232641,0.42913973716004727,0.43218149198157163,0.60060382968157788,0.51945900281713187,0.48399270488872181,0.34829752316801327],[0.30717027549594017,0.38609602365230039,0.58323534255593112,0.33952601695192169,0.76254971423885964,0.56060435629075689,0.6128715011510113,0.36118558255833227,0.80082801254942737,0.3780380801516342,0.32495117289034986],[0.3239079329799639,0.38047559223861399,0.42762312333316288,0.78608707513252618,0.3864605030007125,0.54745615983984064,0.26148608545164437,0.69735366750213912,0.65821368306311967,0.63789096071477192,0.31010129522996993],[0.59031664684391572,0.61281817068577271,0.44517073872984581,0.58764543369952826,0.65951815680520409,0.53526041966487736,0.58091812793971476,0.35502540007848138,0.44649564789548529,0.58151774839503478,0.022369587748604848],[0.56924368573153461,0.17862733255105467,0.80596039173186085,0.72329549579972641,0.51270863965778435,0.38927270986436346,0.3940825489365502,0.47696439307545108,0.30419178209140757,0.47036105501465125,0.59234804403208052],[0.46329954901810694,0.1239194389768859,0.46453230190508804,0.73608864917024164,0.7332861700353942,0.544987330501714,0.50886873310178526,0.24885736017714802,0.5277166703749756,0.55700036622326976,0.50849950900185592],[0.28523457331974839,0.63648607272947599,0.29939392088241484,0.6086504787983541,0.80239144936057849,0.50513345644623908,0.45475359571359802,0.58748024636614349,0.32508327747534088,0.64473260950330069,0.26771639789127089],[0.31642583731166701,0.38818135595462855,0.66922460610636869,0.82020415116009948,0.59769669456287811,0.47336897745049339,0.32106942063572452,0.39554392638329106,0.36655214358378657,0.72510430830086969,0.34368465703665813]],"text":[["row: cg1261<br>column: A8I3<br>value: -1.403246","row: cg1261<br>column: AA2U<br>value:  0.112938","row: cg1261<br>column: A8Y6<br>value: -0.464198","row: cg1261<br>column: AA38<br>value:  0.443472","row: cg1261<br>column: AA30<br>value:  0.880945","row: cg1261<br>column: A95A<br>value: -0.719846","row: cg1261<br>column: AA36<br>value:  1.579517","row: cg1261<br>column: AA2G<br>value:  0.856633","row: cg1261<br>column: A8Y2<br>value:  0.870016","row: cg1261<br>column: AAVB<br>value: -1.105174","row: cg1261<br>column: AA2Q<br>value: -1.051058"],["row: cg1249<br>column: A8I3<br>value:  0.166919","row: cg1249<br>column: AA2U<br>value:  0.662096","row: cg1249<br>column: A8Y6<br>value: -1.591638","row: cg1249<br>column: AA38<br>value:  0.629000","row: cg1249<br>column: AA30<br>value:  1.135595","row: cg1249<br>column: A95A<br>value: -0.022453","row: cg1249<br>column: AA36<br>value: -0.662608","row: cg1249<br>column: AA2G<br>value:  0.172752","row: cg1249<br>column: A8Y2<br>value:  1.082812","row: cg1249<br>column: AAVB<br>value:  0.317623","row: cg1249<br>column: AA2Q<br>value: -1.890099"],["row: cg1245<br>column: A8I3<br>value:  0.311717","row: cg1245<br>column: AA2U<br>value:  2.824726","row: cg1245<br>column: A8Y6<br>value: -0.701061","row: cg1245<br>column: AA38<br>value: -0.375397","row: cg1245<br>column: AA30<br>value: -0.518457","row: cg1245<br>column: A95A<br>value: -0.819441","row: cg1245<br>column: AA36<br>value: -0.269464","row: cg1245<br>column: AA2G<br>value:  0.154640","row: cg1245<br>column: A8Y2<br>value:  0.093787","row: cg1245<br>column: AAVB<br>value: -0.348117","row: cg1245<br>column: AA2Q<br>value: -0.352933"],["row: cg957<br>column: A8I3<br>value: -2.579347","row: cg957<br>column: AA2U<br>value:  0.615653","row: cg957<br>column: A8Y6<br>value: -1.206486","row: cg957<br>column: AA38<br>value:  0.440614","row: cg957<br>column: AA30<br>value:  0.588089","row: cg957<br>column: A95A<br>value:  0.113618","row: cg957<br>column: AA36<br>value:  0.456912","row: cg957<br>column: AA2G<br>value:  0.410243","row: cg957<br>column: A8Y2<br>value:  0.658920","row: cg957<br>column: AAVB<br>value:  0.356480","row: cg957<br>column: AA2Q<br>value:  0.145304"],["row: cg948<br>column: A8I3<br>value:  0.365583","row: cg948<br>column: AA2U<br>value:  0.098891","row: cg948<br>column: A8Y6<br>value:  0.443468","row: cg948<br>column: AA38<br>value:  0.676892","row: cg948<br>column: AA30<br>value:  0.360106","row: cg948<br>column: A95A<br>value: -0.675943","row: cg948<br>column: AA36<br>value: -0.272559","row: cg948<br>column: AA2G<br>value:  0.489597","row: cg948<br>column: A8Y2<br>value:  0.658957","row: cg948<br>column: AAVB<br>value:  0.595801","row: cg948<br>column: AA2Q<br>value: -2.740794"],["row: cg924<br>column: A8I3<br>value:  0.257132","row: cg924<br>column: AA2U<br>value:  0.942735","row: cg924<br>column: A8Y6<br>value: -0.964099","row: cg924<br>column: AA38<br>value: -2.430928","row: cg924<br>column: AA30<br>value:  0.310993","row: cg924<br>column: A95A<br>value: -0.301655","row: cg924<br>column: AA36<br>value: -0.006918","row: cg924<br>column: AA2G<br>value:  0.488019","row: cg924<br>column: A8Y2<br>value:  1.300265","row: cg924<br>column: AAVB<br>value:  0.264822","row: cg924<br>column: AA2Q<br>value:  0.139634"],["row: cg905<br>column: A8I3<br>value:  0.499798","row: cg905<br>column: AA2U<br>value:  1.130437","row: cg905<br>column: A8Y6<br>value: -1.361640","row: cg905<br>column: AA38<br>value: -1.040365","row: cg905<br>column: AA30<br>value:  0.710217","row: cg905<br>column: A95A<br>value: -0.051187","row: cg905<br>column: AA36<br>value: -0.579548","row: cg905<br>column: AA2G<br>value: -0.016017","row: cg905<br>column: A8Y2<br>value:  1.678135","row: cg905<br>column: AAVB<br>value:  0.353629","row: cg905<br>column: AA2Q<br>value: -1.323459"],["row: cg769<br>column: A8I3<br>value: -0.742554","row: cg769<br>column: AA2U<br>value:  1.777287","row: cg769<br>column: A8Y6<br>value: -0.066851","row: cg769<br>column: AA38<br>value: -1.239706","row: cg769<br>column: AA30<br>value: -0.152076","row: cg769<br>column: A95A<br>value: -0.294342","row: cg769<br>column: AA36<br>value: -0.106917","row: cg769<br>column: AA2G<br>value:  1.951927","row: cg769<br>column: A8Y2<br>value: -0.721261","row: cg769<br>column: AAVB<br>value:  0.125019","row: cg769<br>column: AA2Q<br>value: -0.530527"],["row: cg734<br>column: A8I3<br>value: -0.303291","row: cg734<br>column: AA2U<br>value:  1.906492","row: cg734<br>column: A8Y6<br>value: -0.767270","row: cg734<br>column: AA38<br>value: -0.210822","row: cg734<br>column: AA30<br>value: -0.474025","row: cg734<br>column: A95A<br>value: -0.716636","row: cg734<br>column: AA36<br>value: -0.729199","row: cg734<br>column: AA2G<br>value:  1.613523","row: cg734<br>column: A8Y2<br>value:  0.964005","row: cg734<br>column: AAVB<br>value: -0.555847","row: cg734<br>column: AA2Q<br>value: -0.726931"],["row: cg721<br>column: A8I3<br>value:  0.699331","row: cg721<br>column: AA2U<br>value: -2.730013","row: cg721<br>column: A8Y6<br>value:  0.276199","row: cg721<br>column: AA38<br>value:  0.628406","row: cg721<br>column: AA30<br>value:  0.744158","row: cg721<br>column: A95A<br>value:  0.696783","row: cg721<br>column: AA36<br>value:  0.329904","row: cg721<br>column: AA2G<br>value: -0.318931","row: cg721<br>column: A8Y2<br>value:  0.045815","row: cg721<br>column: AAVB<br>value:  0.178722","row: cg721<br>column: AA2Q<br>value: -0.550375"],["row: cg714<br>column: A8I3<br>value: -0.581236","row: cg714<br>column: AA2U<br>value:  0.982764","row: cg714<br>column: A8Y6<br>value:  0.296222","row: cg714<br>column: AA38<br>value: -0.613385","row: cg714<br>column: AA30<br>value:  2.473092","row: cg714<br>column: A95A<br>value: -0.137053","row: cg714<br>column: AA36<br>value: -0.333842","row: cg714<br>column: AA2G<br>value:  0.121838","row: cg714<br>column: A8Y2<br>value: -0.720335","row: cg714<br>column: AAVB<br>value: -0.328682","row: cg714<br>column: AA2Q<br>value: -1.159384"],["row: cg658<br>column: A8I3<br>value:  0.625509","row: cg658<br>column: AA2U<br>value: -2.058508","row: cg658<br>column: A8Y6<br>value:  0.094334","row: cg658<br>column: AA38<br>value: -0.338381","row: cg658<br>column: AA30<br>value:  0.143008","row: cg658<br>column: A95A<br>value:  0.547298","row: cg658<br>column: AA36<br>value: -1.512025","row: cg658<br>column: AA2G<br>value:  0.895433","row: cg658<br>column: A8Y2<br>value:  1.302718","row: cg658<br>column: AAVB<br>value:  0.371233","row: cg658<br>column: AA2Q<br>value: -0.070618"],["row: cg622<br>column: A8I3<br>value:  0.016484","row: cg622<br>column: AA2U<br>value:  2.727110","row: cg622<br>column: A8Y6<br>value: -0.732105","row: cg622<br>column: AA38<br>value: -0.553179","row: cg622<br>column: AA30<br>value: -0.673107","row: cg622<br>column: A95A<br>value: -0.352408","row: cg622<br>column: AA36<br>value: -0.335479","row: cg622<br>column: AA2G<br>value:  0.601879","row: cg622<br>column: A8Y2<br>value:  0.150265","row: cg622<br>column: AAVB<br>value: -0.047123","row: cg622<br>column: AA2Q<br>value: -0.802337"],["row: cg363<br>column: A8I3<br>value: -1.031232","row: cg363<br>column: AA2U<br>value: -0.591969","row: cg363<br>column: A8Y6<br>value:  0.505214","row: cg363<br>column: AA38<br>value: -0.851155","row: cg363<br>column: AA30<br>value:  1.503191","row: cg363<br>column: A95A<br>value:  0.379261","row: cg363<br>column: AA36<br>value:  0.670154","row: cg363<br>column: AA2G<br>value: -0.730608","row: cg363<br>column: A8Y2<br>value:  1.716230","row: cg363<br>column: AAVB<br>value: -0.636815","row: cg363<br>column: AA2Q<br>value: -0.932272"],["row: cg321<br>column: A8I3<br>value: -0.938078","row: cg321<br>column: AA2U<br>value: -0.623249","row: cg321<br>column: A8Y6<br>value: -0.360849","row: cg321<br>column: AA38<br>value:  1.634189","row: cg321<br>column: AA30<br>value: -0.589940","row: cg321<br>column: A95A<br>value:  0.306084","row: cg321<br>column: AA36<br>value: -1.285488","row: cg321<br>column: AA2G<br>value:  1.140342","row: cg321<br>column: A8Y2<br>value:  0.922507","row: cg321<br>column: AAVB<br>value:  0.809401","row: cg321<br>column: AA2Q<br>value: -1.014919"],["row: cg292<br>column: A8I3<br>value:  0.544625","row: cg292<br>column: AA2U<br>value:  0.669858","row: cg292<br>column: A8Y6<br>value: -0.263187","row: cg292<br>column: AA38<br>value:  0.529758","row: cg292<br>column: AA30<br>value:  0.929767","row: cg292<br>column: A95A<br>value:  0.238209","row: cg292<br>column: AA36<br>value:  0.492317","row: cg292<br>column: AA2G<br>value: -0.764893","row: cg292<br>column: A8Y2<br>value: -0.255813","row: cg292<br>column: AAVB<br>value:  0.495655","row: cg292<br>column: AA2Q<br>value: -2.616295"],["row: cg289<br>column: A8I3<br>value:  0.427343","row: cg289<br>column: AA2U<br>value: -1.746640","row: cg289<br>column: A8Y6<br>value:  1.744794","row: cg289<br>column: AA38<br>value:  1.284721","row: cg289<br>column: AA30<br>value:  0.112696","row: cg289<br>column: A95A<br>value: -0.574289","row: cg289<br>column: AA36<br>value: -0.547520","row: cg289<br>column: AA2G<br>value: -0.086239","row: cg289<br>column: A8Y2<br>value: -1.047808","row: cg289<br>column: AAVB<br>value: -0.122990","row: cg289<br>column: AA2Q<br>value:  0.555931"],["row: cg236<br>column: A8I3<br>value: -0.162291","row: cg236<br>column: AA2U<br>value: -2.051118","row: cg236<br>column: A8Y6<br>value: -0.155430","row: cg236<br>column: AA38<br>value:  1.355922","row: cg236<br>column: AA30<br>value:  1.340325","row: cg236<br>column: A95A<br>value:  0.292344","row: cg236<br>column: AA36<br>value:  0.091325","row: cg236<br>column: AA2G<br>value: -1.355773","row: cg236<br>column: A8Y2<br>value:  0.196224","row: cg236<br>column: AAVB<br>value:  0.359203","row: cg236<br>column: AA2Q<br>value:  0.089270"],["row: cg165<br>column: A8I3<br>value: -1.153315","row: cg165<br>column: AA2U<br>value:  0.801582","row: cg165<br>column: A8Y6<br>value: -1.074511","row: cg165<br>column: AA38<br>value:  0.646662","row: cg165<br>column: AA30<br>value:  1.724931","row: cg165<br>column: A95A<br>value:  0.070536","row: cg165<br>column: AA36<br>value: -0.209854","row: cg165<br>column: AA2G<br>value:  0.528839","row: cg165<br>column: A8Y2<br>value: -0.931536","row: cg165<br>column: AAVB<br>value:  0.847478","row: cg165<br>column: AA2Q<br>value: -1.250813"],["row: cg29<br>column: A8I3<br>value: -0.979720","row: cg29<br>column: AA2U<br>value: -0.580363","row: cg29<br>column: A8Y6<br>value:  0.983789","row: cg29<br>column: AA38<br>value:  1.824068","row: cg29<br>column: AA30<br>value:  0.585699","row: cg29<br>column: A95A<br>value: -0.106249","row: cg29<br>column: AA36<br>value: -0.953876","row: cg29<br>column: AA2G<br>value: -0.539386","row: cg29<br>column: A8Y2<br>value: -0.700741","row: cg29<br>column: AAVB<br>value:  1.294788","row: cg29<br>column: AA2Q<br>value: -0.828010"]],"colorscale":[[0,"#440154"],[0.00193704545376707,"#440155"],[0.022369587748604848,"#46095D"],[0.02900839400057692,"#460C5F"],[0.055676003152525667,"#481668"],[0.12259152688283283,"#472C7A"],[0.1239194389768859,"#472D7B"],[0.15285089103088489,"#453581"],[0.17862733255105467,"#433E85"],[0.2064777384215912,"#404688"],[0.22078231063140308,"#3E4989"],[0.24032761647023457,"#3C4F8A"],[0.24780322387554587,"#3B518B"],[0.24885736017714802,"#3B518B"],[0.25466357659249289,"#3A538B"],[0.26148608545164437,"#39558C"],[0.26771639789127089,"#39578C"],[0.26971203181473496,"#38588C"],[0.27568097744803011,"#38598C"],[0.28414418396262969,"#375B8D"],[0.28523457331974839,"#365C8D"],[0.29388446383518507,"#355E8D"],[0.29939392088241484,"#355F8D"],[0.30360790792530157,"#34608D"],[0.30419178209140757,"#34618D"],[0.30552923633205914,"#34618D"],[0.30717027549594017,"#34618D"],[0.31010129522996993,"#33628D"],[0.31642583731166701,"#32648E"],[0.31923256446027082,"#32648E"],[0.32106942063572452,"#32658E"],[0.3239079329799639,"#31668E"],[0.32495117289034986,"#31668E"],[0.32508327747534088,"#31668E"],[0.33952601695192169,"#306A8E"],[0.34368465703665813,"#2F6B8E"],[0.34522436541840146,"#2F6B8E"],[0.34829752316801327,"#2F6C8E"],[0.35459820551326238,"#2E6D8E"],[0.35502540007848138,"#2E6E8E"],[0.3590391901680825,"#2E6F8E"],[0.36091666286277885,"#2E6F8E"],[0.36118558255833227,"#2E6F8E"],[0.36143883478522787,"#2E6F8E"],[0.36184639071541996,"#2E6F8E"],[0.36286503174613477,"#2D708E"],[0.36303144309449081,"#2D708E"],[0.36311924134040191,"#2D708E"],[0.36369606396023796,"#2D708E"],[0.36649454132138393,"#2D708E"],[0.36655214358378657,"#2D708E"],[0.3710077205170606,"#2C718E"],[0.37151730667232641,"#2C718E"],[0.3734036871649084,"#2C718E"],[0.3780380801516342,"#2C728E"],[0.38047559223861399,"#2C738E"],[0.38224794483410057,"#2C738E"],[0.38609602365230039,"#2B748E"],[0.3864605030007125,"#2B758E"],[0.38802438131416817,"#2B758E"],[0.38818135595462855,"#2B758E"],[0.38832776324971624,"#2B758E"],[0.38927270986436346,"#2B758E"],[0.39258636162743699,"#2A768E"],[0.39306561722572991,"#2A768E"],[0.39356950279435754,"#2A768E"],[0.3940825489365502,"#2A768E"],[0.39554392638329106,"#2A778E"],[0.39713573498250787,"#2A778E"],[0.39930452634906621,"#2A788E"],[0.40728793810910557,"#297A8E"],[0.40905353460669053,"#297A8E"],[0.42500921386934232,"#277E8E"],[0.42762312333316288,"#277F8E"],[0.4290453939069887,"#277F8E"],[0.42913973716004727,"#277F8E"],[0.42991068942045263,"#27808E"],[0.43166012789195823,"#27808E"],[0.43218149198157163,"#27808E"],[0.43247569540782183,"#27808E"],[0.43340288512517222,"#26818E"],[0.43515476811970877,"#26818E"],[0.43796501340333915,"#26828E"],[0.43825899900381132,"#26828E"],[0.4395729933845185,"#26828E"],[0.44348677920443591,"#26828E"],[0.44404295808279942,"#26828E"],[0.44517073872984581,"#25838E"],[0.44649564789548529,"#25838E"],[0.45457968569724,"#25858E"],[0.45475359571359802,"#25858E"],[0.46329954901810694,"#24878E"],[0.46453230190508804,"#24878E"],[0.4651350247861894,"#23888E"],[0.46783420019859351,"#23888E"],[0.47036105501465125,"#23898E"],[0.47324903980290561,"#238A8D"],[0.47336897745049339,"#238A8D"],[0.47696439307545108,"#228B8D"],[0.47977107439642769,"#228B8D"],[0.48044803445207707,"#228C8D"],[0.48326248177255754,"#228C8D"],[0.48399270488872181,"#228C8D"],[0.48842541524304051,"#218E8D"],[0.48958176439991691,"#218E8D"],[0.4912166421033724,"#218E8D"],[0.49542152325486821,"#218F8D"],[0.50069166070164373,"#21918C"],[0.50513345644623908,"#20928C"],[0.50849950900185592,"#20928C"],[0.50886873310178526,"#20928C"],[0.50931102978828091,"#20928C"],[0.50940942818080159,"#20928C"],[0.5102281498555209,"#20928C"],[0.51270863965778435,"#20938C"],[0.51275214195658569,"#20938C"],[0.51287418780426308,"#20938C"],[0.51435128750973691,"#20938C"],[0.51492273880008088,"#20938C"],[0.51754883346767722,"#1F948C"],[0.51815493750737096,"#1F948C"],[0.518567478953467,"#1F948C"],[0.51945900281713187,"#1F948C"],[0.52024504919170567,"#1F958B"],[0.52245121753715362,"#1F958B"],[0.52349928239080457,"#1F958B"],[0.52457204616389408,"#1F968B"],[0.5277166703749756,"#1F978B"],[0.53526041966487736,"#1F988B"],[0.53866053360517663,"#1F998A"],[0.54004220739882458,"#1F9A8A"],[0.54208644302699416,"#1F9A8A"],[0.544987330501714,"#1E9B8A"],[0.54568419811285929,"#1E9B8A"],[0.54745615983984064,"#1E9C89"],[0.54833816530234358,"#1E9C89"],[0.54846831113804384,"#1E9C89"],[0.54952943228555429,"#1E9C89"],[0.55173606197487735,"#1E9D89"],[0.55599884083409346,"#1F9E89"],[0.55651109227645612,"#1F9E89"],[0.55700036622326976,"#1F9E89"],[0.55716268115555068,"#1F9E89"],[0.55814673623578637,"#1F9E89"],[0.55916200585568188,"#1F9F88"],[0.56060435629075689,"#1F9F88"],[0.56617126286932651,"#1FA088"],[0.56924368573153461,"#1FA188"],[0.57162814203880963,"#1FA187"],[0.57214097403836328,"#1FA187"],[0.57214175749949037,"#1FA187"],[0.57455663765015497,"#1FA287"],[0.58014573465145958,"#20A386"],[0.58042940126397202,"#20A386"],[0.58091812793971476,"#20A386"],[0.58151774839503478,"#20A386"],[0.58226227517761431,"#20A386"],[0.58323534255593112,"#20A486"],[0.58748024636614349,"#21A585"],[0.58764543369952826,"#21A585"],[0.59031664684391572,"#21A685"],[0.59079686334286419,"#21A685"],[0.59234804403208052,"#21A685"],[0.59769669456287811,"#22A785"],[0.5981261277663833,"#22A884"],[0.59951190742659766,"#22A884"],[0.60060382968157788,"#22A884"],[0.60307882320262318,"#23A983"],[0.60484966817768382,"#23A983"],[0.60537029785461161,"#23A983"],[0.605477036610791,"#23A983"],[0.6086504787983541,"#24AA83"],[0.61085295447637422,"#25AB82"],[0.61085964344396981,"#25AB82"],[0.61142362592355703,"#25AB82"],[0.61281817068577271,"#25AB82"],[0.6128715011510113,"#25AB82"],[0.6140820853452077,"#25AC82"],[0.61765603126664559,"#26AD81"],[0.61811393627451139,"#26AD81"],[0.62006979717518573,"#26AD81"],[0.62616828485545384,"#28AE80"],[0.63648607272947599,"#2BB07F"],[0.63789096071477192,"#2BB17E"],[0.64473260950330069,"#2DB27D"],[0.64637754626334265,"#2EB37C"],[0.64878224525868433,"#2EB37C"],[0.65074584812695158,"#2FB47C"],[0.65334898659937868,"#30B57B"],[0.65821368306311967,"#32B67A"],[0.65951815680520409,"#32B67A"],[0.66184812716107755,"#34B679"],[0.66566994436591953,"#35B779"],[0.66904051023046629,"#36B878"],[0.66922460610636869,"#36B878"],[0.68701685808124435,"#3DBC74"],[0.69557401807822961,"#41BD72"],[0.69650089379693514,"#41BE71"],[0.69735366750213912,"#42BE71"],[0.72329549579972641,"#4FC36B"],[0.72510430830086969,"#50C46A"],[0.72608826817992578,"#50C46A"],[0.72652914902006205,"#51C46A"],[0.7332861700353942,"#54C568"],[0.73608864917024164,"#55C667"],[0.76254971423885964,"#64CB5F"],[0.77626377520359702,"#6CCD5A"],[0.78237375005061338,"#6FCF57"],[0.78608707513252618,"#71CF57"],[0.79398310099905334,"#76D054"],[0.80082801254942737,"#7AD151"],[0.80239144936057849,"#7BD250"],[0.80596039173186085,"#7ED34F"],[0.81179867066879563,"#81D34D"],[0.82020415116009948,"#86D549"],[0.83501389025866235,"#90D743"],[0.84317758788043784,"#95D840"],[0.93681934869642591,"#D5E21A"],[0.98246067877369792,"#F3E61E"],[1,"#FDE725"]],"type":"heatmap","showscale":false,"autocolorscale":false,"showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5,null,0.5,11.5],"y":[0.5,0.5,null,1.5,1.5,null,10.5,10.5,null,11.5,11.5,null,12.5,12.5,null,13.5,13.5,null,14.5,14.5,null,15.5,15.5,null,16.5,16.5,null,17.5,17.5,null,18.5,18.5,null,19.5,19.5,null,2.5,2.5,null,20.5,20.5,null,3.5,3.5,null,4.5,4.5,null,5.5,5.5,null,6.5,6.5,null,7.5,7.5,null,8.5,8.5,null,9.5,9.5],"text":"","type":"scatter","mode":"lines","line":{"width":1.8897637795275593,"color":"rgba(255,255,255,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null,"marker":{"showscale":false}},{"x":[0.5,0.5,null,1.5,1.5,null,10.5,10.5,null,11.5,11.5,null,2.5,2.5,null,3.5,3.5,null,4.5,4.5,null,5.5,5.5,null,6.5,6.5,null,7.5,7.5,null,8.5,8.5,null,9.5,9.5],"y":[0.5,20.5,null,0.5,20.5,null,0.5,20.5,null,0.5,20.5,null,0.5,20.5,null,0.5,20.5,null,0.5,20.5,null,0.5,20.5,null,0.5,20.5,null,0.5,20.5,null,0.5,20.5,null,0.5,20.5],"text":"","type":"scatter","mode":"lines","line":{"width":1.8897637795275593,"color":"rgba(255,255,255,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null,"marker":{"showscale":false}},{"x":[1],"y":[1],"name":"76e9dc1c380a49fd747e80bbbc4d66c6","type":"scatter","mode":"markers","opacity":0,"hoverinfo":"skip","showlegend":false,"marker":{"color":[0,1],"colorscale":[[0,"#440154"],[0.0033444816053511931,"#440256"],[0.0066889632107023063,"#450357"],[0.0100334448160535,"#450558"],[0.013377926421404692,"#450659"],[0.016722408026755887,"#46075B"],[0.020066889632107,"#46085C"],[0.023411371237458192,"#460A5D"],[0.026755852842809385,"#460B5E"],[0.030100334448160498,"#470C5F"],[0.03344481605351169,"#470E61"],[0.036789297658862886,"#470F62"],[0.040133779264214076,"#471063"],[0.043478260869565188,"#471164"],[0.046822742474916385,"#471365"],[0.050167224080267574,"#481467"],[0.053511705685618687,"#481568"],[0.056856187290969883,"#481768"],[0.060200668896321072,"#481769"],[0.063545150501672268,"#48186A"],[0.066889632107023381,"#481A6C"],[0.070234113712374577,"#481B6D"],[0.073578595317725773,"#481C6E"],[0.076923076923076886,"#481D6F"],[0.080267558528428068,"#481E6F"],[0.083612040133779264,"#481F70"],[0.08695652173913046,"#482071"],[0.090301003344481656,"#482173"],[0.093645484949832769,"#482374"],[0.096989966555183965,"#482475"],[0.10033444816053508,"#482576"],[0.10367892976588626,"#482576"],[0.10702341137123746,"#482777"],[0.11036789297658865,"#482878"],[0.11371237458193985,"#482979"],[0.11705685618729096,"#472A7A"],[0.12040133779264214,"#472B7A"],[0.12374581939799326,"#472D7B"],[0.12709030100334445,"#472D7B"],[0.13043478260869565,"#472E7C"],[0.13377926421404684,"#472F7D"],[0.13712374581939799,"#46307E"],[0.14046822742474915,"#46327E"],[0.14381270903010035,"#46337F"],[0.14715719063545149,"#463480"],[0.15050167224080266,"#463480"],[0.15384615384615383,"#453581"],[0.15719063545150502,"#453781"],[0.16053511705685619,"#453882"],[0.16387959866220733,"#443983"],[0.16722408026755853,"#443A83"],[0.17056856187290972,"#443A83"],[0.17391304347826086,"#443C84"],[0.17725752508361203,"#433D84"],[0.18060200668896323,"#433E85"],[0.18394648829431437,"#423F85"],[0.18729096989966554,"#424086"],[0.19063545150501673,"#424186"],[0.19397993311036787,"#424186"],[0.19732441471571907,"#414387"],[0.20066889632107021,"#414487"],[0.20401337792642141,"#404588"],[0.20735785953177258,"#404688"],[0.21070234113712377,"#3F4788"],[0.21404682274247491,"#3F4889"],[0.21739130434782608,"#3F4889"],[0.22073578595317725,"#3E4989"],[0.22408026755852842,"#3E4A89"],[0.22742474916387961,"#3E4C8A"],[0.23076923076923075,"#3D4D8A"],[0.23411371237458195,"#3D4E8A"],[0.23745819397993312,"#3C4F8A"],[0.24080267558528429,"#3C4F8A"],[0.24414715719063546,"#3C508B"],[0.2474916387959866,"#3B518B"],[0.25083612040133779,"#3B528B"],[0.25418060200668896,"#3A538B"],[0.25752508361204013,"#3A548C"],[0.2608695652173913,"#39558C"],[0.26421404682274247,"#39558C"],[0.26755852842809363,"#39568C"],[0.2709030100334448,"#38588C"],[0.27424749163879597,"#38598C"],[0.27759197324414714,"#375A8C"],[0.28093645484949836,"#375B8D"],[0.28428093645484948,"#375B8D"],[0.2876254180602007,"#365C8D"],[0.29096989966555181,"#365D8D"],[0.29431438127090298,"#355E8D"],[0.2976588628762542,"#355F8D"],[0.30100334448160532,"#34608D"],[0.30434782608695654,"#34618D"],[0.30769230769230765,"#34618D"],[0.31103678929765888,"#33628D"],[0.31438127090301005,"#33638D"],[0.31772575250836121,"#32648E"],[0.32107023411371238,"#32658E"],[0.32441471571906355,"#31668E"],[0.32775919732441472,"#31678E"],[0.33110367892976589,"#31678E"],[0.33444816053511706,"#31688E"],[0.33779264214046822,"#30698E"],[0.34113712374581939,"#306A8E"],[0.34448160535117056,"#2F6B8E"],[0.34782608695652173,"#2F6C8E"],[0.3511705685618729,"#2E6D8E"],[0.35451505016722407,"#2E6D8E"],[0.35785953177257523,"#2E6E8E"],[0.36120401337792646,"#2E6F8E"],[0.36454849498327757,"#2D708E"],[0.36789297658862874,"#2D718E"],[0.37123745819397996,"#2C718E"],[0.37458193979933108,"#2C728E"],[0.37792642140468224,"#2C728E"],[0.38127090301003347,"#2C738E"],[0.38461538461538464,"#2B748E"],[0.38795986622073575,"#2B758E"],[0.39130434782608697,"#2A768E"],[0.39464882943143814,"#2A778E"],[0.39799331103678931,"#2A778E"],[0.40133779264214042,"#2A788E"],[0.40468227424749165,"#29798E"],[0.40802675585284282,"#297A8E"],[0.41137123745819393,"#297B8E"],[0.41471571906354515,"#287C8E"],[0.41806020066889632,"#287D8E"],[0.42140468227424754,"#287D8E"],[0.42474916387959866,"#277E8E"],[0.42809364548494983,"#277F8E"],[0.43143812709030105,"#27808E"],[0.43478260869565216,"#26818E"],[0.43812709030100333,"#26828E"],[0.4414715719063545,"#26828E"],[0.44481605351170572,"#26828E"],[0.44816053511705684,"#25838E"],[0.451505016722408,"#25848E"],[0.45484949832775923,"#25858E"],[0.45819397993311034,"#24868E"],[0.46153846153846151,"#24878E"],[0.46488294314381273,"#23888E"],[0.4682274247491639,"#23888E"],[0.47157190635451501,"#23898E"],[0.47491638795986624,"#238A8D"],[0.47826086956521741,"#228B8D"],[0.48160535117056857,"#228C8D"],[0.48494983277591969,"#228D8D"],[0.48829431438127091,"#218E8D"],[0.49163879598662208,"#218E8D"],[0.49498327759197319,"#218F8D"],[0.49832775919732442,"#21908D"],[0.50167224080267558,"#21918C"],[0.50501672240802675,"#20928C"],[0.50836120401337792,"#20928C"],[0.51170568561872909,"#20928C"],[0.51505016722408026,"#20938C"],[0.51839464882943143,"#1F948C"],[0.52173913043478259,"#1F958B"],[0.52508361204013376,"#1F968B"],[0.52842809364548493,"#1F978B"],[0.5317725752508361,"#1F988B"],[0.53511705685618727,"#1F988B"],[0.53846153846153844,"#1F998A"],[0.5418060200668896,"#1F9A8A"],[0.54515050167224077,"#1E9B8A"],[0.54849498327759194,"#1E9C89"],[0.55183946488294322,"#1E9D89"],[0.55518394648829428,"#1F9E89"],[0.55852842809364545,"#1F9E89"],[0.56187290969899673,"#1F9F88"],[0.56521739130434778,"#1FA088"],[0.56856187290969895,"#1FA188"],[0.57190635451505012,"#1FA187"],[0.5752508361204014,"#1FA287"],[0.57859531772575246,"#20A386"],[0.58193979933110362,"#20A386"],[0.5852842809364549,"#20A486"],[0.58862876254180596,"#21A585"],[0.59197324414715713,"#21A685"],[0.59531772575250841,"#22A785"],[0.59866220735785958,"#22A884"],[0.60200668896321063,"#23A983"],[0.60535117056856191,"#23A983"],[0.60869565217391308,"#24AA83"],[0.61204013377926425,"#25AB82"],[0.61538461538461531,"#25AC82"],[0.61872909698996659,"#26AD81"],[0.62207357859531776,"#27AD81"],[0.62541806020066881,"#27AD81"],[0.62876254180602009,"#28AE80"],[0.63210702341137126,"#29AF7F"],[0.63545150501672243,"#2AB07F"],[0.6387959866220736,"#2CB17E"],[0.64214046822742477,"#2DB27D"],[0.64548494983277593,"#2EB37C"],[0.6488294314381271,"#2EB37C"],[0.65217391304347827,"#30B47C"],[0.65551839464882944,"#31B57B"],[0.65886287625418061,"#32B67A"],[0.66220735785953178,"#34B679"],[0.66555183946488294,"#35B779"],[0.66889632107023411,"#36B878"],[0.67224080267558528,"#37B878"],[0.67558528428093645,"#39B977"],[0.67892976588628762,"#3ABA76"],[0.68227424749163879,"#3BBB75"],[0.68561872909698995,"#3DBC74"],[0.68896321070234112,"#3EBC73"],[0.69230769230769229,"#40BD72"],[0.69565217391304346,"#41BD72"],[0.69899665551839463,"#42BE71"],[0.7023411371237458,"#44BF70"],[0.70568561872909696,"#46C06F"],[0.70903010033444813,"#48C16E"],[0.7123745819397993,"#49C16D"],[0.71571906354515047,"#4BC26C"],[0.71906354515050175,"#4DC26C"],[0.72240802675585292,"#4EC36B"],[0.72575250836120397,"#50C46A"],[0.72909698996655514,"#52C569"],[0.73244147157190631,"#54C568"],[0.73578595317725748,"#55C667"],[0.73913043478260865,"#57C666"],[0.74247491638795993,"#59C765"],[0.7458193979933111,"#5AC864"],[0.74916387959866215,"#5CC863"],[0.75250836120401332,"#5EC962"],[0.75585284280936449,"#5FCA61"],[0.75919732441471566,"#62CB5F"],[0.76254180602006694,"#64CB5F"],[0.76588628762541811,"#66CB5D"],[0.76923076923076927,"#67CC5C"],[0.77257525083612044,"#69CD5B"],[0.7759197324414715,"#6CCD5A"],[0.77926421404682267,"#6DCE59"],[0.78260869565217395,"#6FCF57"],[0.78595317725752512,"#71CF57"],[0.78929765886287628,"#74D055"],[0.79264214046822745,"#75D054"],[0.79598662207357862,"#77D153"],[0.79933110367892968,"#79D151"],[0.80267558528428085,"#7BD250"],[0.80602006688963213,"#7ED34F"],[0.80936454849498329,"#80D34E"],[0.81270903010033446,"#82D34D"],[0.81605351170568563,"#84D44B"],[0.81939799331103691,"#86D549"],[0.82274247491638786,"#88D548"],[0.82608695652173914,"#8AD647"],[0.8294314381270903,"#8DD645"],[0.83277591973244147,"#8FD644"],[0.83612040133779264,"#91D743"],[0.83946488294314381,"#93D741"],[0.84280936454849509,"#95D840"],[0.84615384615384603,"#97D83E"],[0.84949832775919731,"#9AD93D"],[0.85284280936454848,"#9CD93C"],[0.85618729096989965,"#9ED93A"],[0.85953177257525082,"#A0DA39"],[0.8628762541806021,"#A2DA37"],[0.86622073578595327,"#A5DB36"],[0.86956521739130432,"#A7DB35"],[0.87290969899665549,"#A9DC33"],[0.87625418060200666,"#ABDC31"],[0.87959866220735783,"#AEDC30"],[0.882943143812709,"#B0DD2F"],[0.88628762541806028,"#B2DD2D"],[0.88963210702341144,"#B5DE2B"],[0.8929765886287625,"#B7DE2A"],[0.89632107023411367,"#B9DE28"],[0.89966555183946484,"#BBDE27"],[0.90301003344481601,"#BEDF26"],[0.90635451505016729,"#C0DF25"],[0.90969899665551845,"#C2DF23"],[0.91304347826086962,"#C4E021"],[0.91638795986622068,"#C7E020"],[0.91973244147157185,"#C9E11F"],[0.92307692307692302,"#CBE11E"],[0.92642140468227419,"#CEE11D"],[0.92976588628762546,"#D0E11C"],[0.93311036789297663,"#D2E21B"],[0.9364548494983278,"#D4E21A"],[0.93979933110367886,"#D7E219"],[0.94314381270903003,"#D9E319"],[0.9464882943143812,"#DBE319"],[0.94983277591973247,"#DDE318"],[0.95317725752508364,"#DFE318"],[0.95652173913043481,"#E2E418"],[0.95986622073578598,"#E4E419"],[0.96321070234113715,"#E6E419"],[0.96655518394648821,"#E8E419"],[0.96989966555183937,"#EBE51A"],[0.97324414715719065,"#EDE51B"],[0.97658862876254182,"#EFE51C"],[0.97993311036789299,"#F1E51D"],[0.98327759197324416,"#F3E61E"],[0.98662207357859544,"#F5E61F"],[0.98996655518394638,"#F7E620"],[0.99331103678929766,"#F9E622"],[0.99665551839464883,"#FBE723"],[1,"#FDE725"]],"colorbar":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"thickness":23.039999999999996,"title":null,"titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"tickmode":"array","ticktext":["-2","-1","0","1","2"],"tickvals":[0.13432711394066513,0.31340594598055266,0.49248477802044027,0.67156361006032794,0.85064244210021533],"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"ticklen":2,"len":0.5},"showscale":false},"xaxis":"x","yaxis":"y","frame":null}],"layout":{"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.5,11.5],"tickmode":"array","ticktext":["A8I3","AA2U","A8Y6","AA38","AA30","A95A","AA36","AA2G","A8Y2","AAVB","AA2Q"],"tickvals":[1,2,3,4,5,6,7,7.9999999999999991,9,10,11],"categoryorder":"array","categoryarray":["A8I3","AA2U","A8Y6","AA38","AA30","A95A","AA36","AA2G","A8Y2","AAVB","AA2Q"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":6.6417600664175991},"tickangle":-45,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.5,20.5],"tickmode":"array","ticktext":["cg1261","cg1249","cg1245","cg957","cg948","cg924","cg905","cg769","cg734","cg721","cg714","cg658","cg622","cg363","cg321","cg292","cg289","cg236","cg165","cg29"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20],"categoryorder":"array","categoryarray":["cg1261","cg1249","cg1245","cg957","cg948","cg924","cg905","cg769","cg734","cg721","cg714","cg658","cg622","cg363","cg321","cg292","cg289","cg236","cg165","cg29"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":6.6417600664176026},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","layer":"below","x0":0,"x1":1,"y0":0,"y1":1}],"images":[],"margin":{"t":30,"r":30,"b":50,"l":0},"paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"y":1,"yanchor":"top"},"hovermode":"closest","barmode":"relative","title":""},"attrs":{"26b73a36da74":{"x":{},"y":{},"fill":{},"text":{},"type":"heatmap"},"26b7136fb1b2":{"yintercept":{}},"26b76c4dd6aa":{"xintercept":{}}},"source":"A","config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false,"displaylogo":false,"modeBarButtonsToRemove":["sendDataToCloud","select2d","lasso2d","autoScale2d","hoverClosestCartesian","hoverCompareCartesian","sendDataToCloud"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"subplot":true,"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-heatmaplyggplot2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13: Using heatmaply ggplot2
</figcaption>
</figure>
</div>
</div>
<p>Similarly, the <code>heatmaply</code> package generates interactive heatmaps, allowing for dynamic analysis and enhanced comprehension of the data.</p>
</section>
</section>
</section>
<section id="lattice-heatmaps" class="level3">
<h3 class="anchored" data-anchor-id="lattice-heatmaps">3. lattice Heatmaps</h3>
<p>The <code>lattice</code> package provides the <code>levelplot()</code> function for constructing heatmaps.</p>
<section id="basic-usage-of-levelplot" class="level4">
<h4 class="anchored" data-anchor-id="basic-usage-of-levelplot">3.1 Basic Usage of <code>levelplot()</code></h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the Sample and Composite columns from characters to factors</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>methylation_long_standardized<span class="sc">$</span>Sample <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(methylation_long_standardized<span class="sc">$</span>Sample)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>methylation_long_standardized<span class="sc">$</span>Composite <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(methylation_long_standardized<span class="sc">$</span>Composite) </span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="fu">levelplot</span>(Standardized_Level <span class="sc">~</span> Sample <span class="sc">*</span> Composite, <span class="at">data =</span> methylation_long_standardized,</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">xlab =</span> <span class="st">"Sample"</span>, <span class="at">ylab =</span> <span class="st">"Composite"</span>,</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>          <span class="at">main =</span> <span class="st">"Heatmap of Standardized TCGA-CHOL Methylation Levels"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-BasicHeatlattice" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-BasicHeatlattice-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-BasicHeatlattice-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-BasicHeatlattice-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14: Basic Heatmap lattice
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap illustrates the distribution of standardized methylation levels between samples and composites in the TCGA-CHOL dataset, highlighting variations in methylation states across different samples.</p>
</section>
<section id="wide-format-matrix-input" class="level4">
<h4 class="anchored" data-anchor-id="wide-format-matrix-input">3.2 Wide-Format Matrix Input</h4>
<p>Unlike the earlier example with long-format data, this approach uses a square matrix.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<p>Rows and columns are flipped in the heatmap.</p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">levelplot</span>(standardized_methylation_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-Wide-FormatMatrixInputlattice" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-Wide-FormatMatrixInputlattice-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-Wide-FormatMatrixInputlattice-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-Wide-FormatMatrixInputlattice-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;15: Wide-Format Matrix Input lattice
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap demonstrates variations in methylation states across TCGA-CHOL samples using a wide-format matrix in <code>levelplot()</code>.</p>
</section>
<section id="flipping-axes" class="level4">
<h4 class="anchored" data-anchor-id="flipping-axes">3.3 Flipping Axes</h4>
<p>R’s <code>t()</code> function transposes the input matrix, flipping the x and y coordinates. This makes the heatmap’s structure align precisely with the input matrix.</p>
<p>Setting <code>aspect = "fill"</code> adjusts the aspect ratio automatically to reflect the matrix’s actual proportions.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Flip X and Y axes</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levelplot</span>( <span class="fu">t</span>(standardized_methylation_matrix),</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">col.regions=</span><span class="fu">heat.colors</span>(<span class="dv">220</span>),</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">aspect =</span> <span class="st">"fill"</span>) <span class="co"># Automatically adjust aspect ratio</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-FlippingAxeslattice" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-FlippingAxeslattice-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-FlippingAxeslattice-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-FlippingAxeslattice-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;16: Flipping Axes lattice
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap reflects the same data as the original matrix with flipped coordinates for structural alignment.</p>
</section>
<section id="color-palettes-2" class="level4">
<h4 class="anchored" data-anchor-id="color-palettes-2">3.4 Color Palettes</h4>
<p>Multiple palette options are available: - Native R palettes: <code>terrain.color()</code>， <code>rainbow()</code>，<code>heat.colors()</code>， <code>topo.colors()</code>，<code>cm.colors()</code> - <code>RColorBrewer</code>palettes - <code>Viridis</code>palettes: viridis, magma, inferno, plasma</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Native color palette</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>p10 <span class="ot">&lt;-</span> <span class="fu">levelplot</span>(volcano, <span class="at">col.regions =</span> <span class="fu">terrain.colors</span>(<span class="dv">100</span>)) <span class="co"># try cm.colors() or terrain.colors()</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>p10</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Using Rcolorbrewer</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>coul <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"PiYG"</span>))(<span class="dv">25</span>)</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>p11 <span class="ot">&lt;-</span> <span class="fu">levelplot</span>(volcano, <span class="at">col.regions =</span> coul)</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>p11</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Using Viridis</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>coul <span class="ot">&lt;-</span> <span class="fu">viridis</span>(<span class="dv">100</span>)</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>p12 <span class="ot">&lt;-</span> <span class="fu">levelplot</span>(volcano, <span class="at">col.regions =</span> coul) </span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>p12</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-ColorPaletteslattice-1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ColorPaletteslattice-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-ColorPaletteslattice-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ColorPaletteslattice-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;17: Color Palettes lattice
</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display">
<div id="fig-ColorPaletteslattice-2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ColorPaletteslattice-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-ColorPaletteslattice-2.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ColorPaletteslattice-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;18: Color Palettes lattice
</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display">
<div id="fig-ColorPaletteslattice-3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ColorPaletteslattice-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-ColorPaletteslattice-3.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ColorPaletteslattice-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;19: Color Palettes lattice
</figcaption>
</figure>
</div>
</div>
</div>
<p>Using the <code>volcano</code> dataset, these heatmaps visualize elevation values and emphasize the impact of palette selection on interpretability.。</p>
</section>
</section>
<section id="pheatmap-heatmaps" class="level3">
<h3 class="anchored" data-anchor-id="pheatmap-heatmaps">4. pheatmap Heatmaps</h3>
<p>The <code>pheatmap</code> package also builds heatmaps and supports data standardization.</p>
<section id="basic-heatmap-2" class="level4">
<h4 class="anchored" data-anchor-id="basic-heatmap-2">4.1 Basic Heatmap</h4>
<div class="sourceCode" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(methylation_matrix_num, <span class="at">scale =</span> <span class="st">"row"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>This heatmap visualizes differences in methylation patterns across samples in the TCGA-CHOL dataset using the <code>pheatmap</code> package.</p>
</section>
<section id="adjusting-dendrograms" class="level4">
<h4 class="anchored" data-anchor-id="adjusting-dendrograms">4.2 Adjusting Dendrograms</h4>
<ul>
<li><code>clustering_distance_rows</code> and <code>clustering_distance_cols</code> represent the distance measures used for row and column clustering. The default is “euclidean”, but it can also be set to “correlation”, meaning clustering will be performed based on the Pearson correlation method.</li>
<li><code>clustering_method</code> allows modification of the clustering method, which can be one of the following: ‘ward’, ‘ward.D’, ‘ward.D2’, ‘single’, ‘complete’, ‘average’, ‘mcquitty’, ‘median’, or ‘centroid’.</li>
</ul>
<div class="sourceCode" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(methylation_matrix_num, <span class="at">scale =</span> <span class="st">"row"</span>, </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"euclidean"</span>, <span class="at">clustering_distance_cols =</span> <span class="st">"euclidean"</span>, <span class="co"># Distance measure used for clustering</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_method =</span> <span class="st">"complete"</span>, <span class="co"># Clustering method</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> T, <span class="at">cluster_cols =</span> F, <span class="co"># Whether to display clustering trees</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">treeheight_row =</span> <span class="dv">30</span>, <span class="at">treeheight_col =</span> <span class="dv">30</span>) <span class="co"># Adjust clustering tree heights</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>This heatmap uses the <code>pheatmap</code> package to display the differences in methylation status across different samples in the TCGA-CHOL dataset, with clustering between samples turned off. The samples are arranged in their original order.</p>
</section>
<section id="angle-adjustment-of-labels-and-segmentation-position-without-clustering" class="level4">
<h4 class="anchored" data-anchor-id="angle-adjustment-of-labels-and-segmentation-position-without-clustering">4.3 Angle Adjustment of Labels and Segmentation Position Without Clustering</h4>
<div class="sourceCode" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(methylation_matrix_num, <span class="at">scale =</span> <span class="st">"row"</span>, </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"euclidean"</span>, <span class="at">clustering_distance_cols =</span> <span class="st">"euclidean"</span>, <span class="co"># Distance measure used for clustering</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_method =</span> <span class="st">"complete"</span>, <span class="co"># Clustering method</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> T, <span class="at">cluster_cols =</span> F, <span class="co"># Whether to display clustering trees</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">treeheight_row =</span> <span class="dv">30</span>, <span class="at">treeheight_col =</span> <span class="dv">30</span>, <span class="co"># Adjust clustering tree heights</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">angle_col =</span> <span class="st">"45"</span>, <span class="co"># Change column label angle</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">gaps_col =</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">6</span>))  <span class="co"># Used only when column clustering is not performed, assuming samples are split into three groups at positions 3 and 6</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Adjusting the label angles in this heatmap can prevent overlapping labels and improve readability. Adjusting the segmentation position under non-clustered conditions can highlight differences between specific samples, facilitating intuitive analysis.</p>
</section>
<section id="adding-values-or-markers" class="level4">
<h4 class="anchored" data-anchor-id="adding-values-or-markers">4.4 Adding Values or Markers</h4>
<div class="sourceCode" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(methylation_matrix_num, <span class="at">scale =</span> <span class="st">"row"</span>,</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="st">"white"</span>, <span class="co"># Change cell border color</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">display_numbers =</span> <span class="cn">TRUE</span>, <span class="co"># Display corresponding values in heatmap cells</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_number =</span> <span class="dv">9</span>, </span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">number_color =</span> <span class="st">"grey30"</span>,</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">number_format =</span> <span class="st">"%.1f"</span>) <span class="co"># Keep one decimal place</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Adding values or markers to the heatmap provides additional context, helping observers quickly identify the specific methylation level of each sample or site, thereby enhancing the intuitiveness and accuracy of data interpretation.</p>
<section id="using-raw-data-for-annotation" class="level5">
<h5 class="anchored" data-anchor-id="using-raw-data-for-annotation"><strong>Using Raw Data for Annotation</strong></h5>
<div class="sourceCode" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(methylation_matrix_num, <span class="at">scale =</span> <span class="st">"row"</span>,</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="st">"white"</span>,</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">display_numbers =</span> <span class="fu">matrix</span>(<span class="fu">ifelse</span>(methylation_matrix_num <span class="sc">&gt;</span> .<span class="dv">8</span> <span class="sc">|</span> methylation_matrix_num <span class="sc">&lt;</span> <span class="sc">-</span>.<span class="dv">8</span>, <span class="st">"*"</span>, <span class="st">""</span>), <span class="at">nrow =</span> <span class="fu">nrow</span>(methylation_matrix_num)),</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_number =</span> <span class="dv">9</span>, </span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">number_color =</span> <span class="st">"grey30"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>This heatmap uses unnormalized data for annotation, making the results incomparable and potentially misleading observers regarding the differences between samples. Unnormalized values may obscure actual relative changes due to absolute value differences between samples, reducing the accuracy of data interpretation.</p>
</section>
<section id="using-normalized-data" class="level5">
<h5 class="anchored" data-anchor-id="using-normalized-data"><strong>Using Normalized Data</strong></h5>
<div class="sourceCode" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(standardized_methylation_matrix, <span class="at">scale =</span> <span class="st">"none"</span>, <span class="co"># No need for further standardization</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="st">"white"</span>,</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">display_numbers =</span> <span class="fu">matrix</span>(<span class="fu">ifelse</span>(standardized_methylation_matrix <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">|</span> standardized_methylation_matrix <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>, <span class="st">"*"</span>, <span class="st">""</span>), <span class="at">nrow =</span> <span class="fu">nrow</span>(standardized_methylation_matrix)),</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_number =</span> <span class="dv">9</span>, </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">number_color =</span> <span class="st">"grey30"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Marking the heatmap with normalized data can intuitively highlight significant values or features, making it easier for observers to recognize notable methylation state changes or outliers, thus providing guidance for subsequent analysis. This approach enhances the readability and interpretability of the data, helping users quickly extract key information.</p>
</section>
</section>
<section id="color-palettes-3" class="level4">
<h4 class="anchored" data-anchor-id="color-palettes-3">4.5 Color Palettes</h4>
<p>The heatmap generated by the <code>pheatmap</code> package allows for various ways to modify the color palette: - Native R palettes:<code>terrain.color()</code>， <code>rainbow()</code>，<code>heat.colors()</code>， <code>topo.colors()</code>，<code>cm.colors()</code> - Custom palettes - <code>RColorBrewer</code> palettes</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Native color palette</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>p13 <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(standardized_methylation_matrix, <span class="at">scale =</span> <span class="st">"none"</span>,</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="st">"white"</span>,</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"navy"</span>,<span class="st">"white"</span>,<span class="st">"firebrick3"</span>))(<span class="dv">100</span>))</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Custom color palette</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>p14 <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(standardized_methylation_matrix, <span class="at">scale =</span> <span class="st">"none"</span>,</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="st">"white"</span>,</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">color =</span> <span class="fu">cm.colors</span>(<span class="dv">100</span>))</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Using Rcolorbrewer</span></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>coul <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"PiYG"</span>))(<span class="dv">25</span>)</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>p15 <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(standardized_methylation_matrix, <span class="at">scale =</span> <span class="st">"none"</span>,</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="st">"white"</span>,</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>         <span class="at">color =</span> coul)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>These three heatmaps use different color palettes to showcase the differences in methylation states among samples in the TCGA-CHOL dataset, emphasizing the importance of palette choice for visual effects and data interpretation.</p>
</section>
</section>
<section id="complexheatmap-heatmaps" class="level3">
<h3 class="anchored" data-anchor-id="complexheatmap-heatmaps">5. complexheatmap Heatmaps</h3>
<p>The <code>ComplexHeatmap</code> package, based on the <code>grid</code> package, uses an object-oriented approach to implement heatmaps and their components. It mainly includes the following classes:</p>
<ul>
<li><code>Heatmap</code>：Draws a single heatmap (note the capitalization).</li>
<li><code>HeatmapList</code>：Draws a list of heatmaps.</li>
<li><code>HeatmapAnnotation</code>: Defines row and column annotations for the heatmap, which can either be part of the heatmap or independent of it.</li>
</ul>
<p>Additionally, there are some internal classes:</p>
<ul>
<li><code>SingleAnnotation</code>：: Defines a single row or column annotation, forming the list elements of <code>HeatmapAnnotation</code>.</li>
<li><code>ColorMapping</code>: Defines the mapping of values to colors.</li>
<li><code>AnnotationFunction</code>: Used to customize annotation graphics.</li>
</ul>
<section id="basic-heatmap-3" class="level4">
<h4 class="anchored" data-anchor-id="basic-heatmap-3">5.1 Basic Heatmap</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(standardized_methylation_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-BasicHeatcomplexheatmap" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-BasicHeatcomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-BasicHeatcomplexheatmap-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-BasicHeatcomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;20: Basic Heatmap complexheatmap
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap uses the <code>ComplexHeatmap</code> package to display the differences in methylation status across different samples in the TCGA-CHOL dataset.</p>
</section>
<section id="color-palettes-4" class="level4">
<h4 class="anchored" data-anchor-id="color-palettes-4">5.2 Color Palettes</h4>
<p>Usually, the matrices we plot consist of continuous data, so the color mapping function needs to take a vector as input and return a vector as output.</p>
<p>The author of this package recommends using the <code>colorRamp2()</code> function from another package, <code>circlize</code>, to set up color mapping. The <code>colorRamp2()</code> function takes two parameters: the first sets the breakpoint values for mapping, and the second specifies the corresponding color values.</p>
<p>In the example below, values between -2 and 2 are linearly interpolated to obtain corresponding colors, with values greater than 2 mapped to red and values less than -2 mapped to green.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>col_fun <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">0</span>, <span class="dv">2</span>), <span class="fu">c</span>(<span class="st">"green"</span>, <span class="st">"white"</span>, <span class="st">"red"</span>))</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(standardized_methylation_matrix, <span class="at">name =</span> <span class="st">"Standardized_Level"</span>, <span class="at">col =</span> col_fun)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-ColorPalettescomplexheatmap" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ColorPalettescomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-ColorPalettescomplexheatmap-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ColorPalettescomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;21: Color Palettes complexheatmap
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap uses <code>colorRamp2()</code> to set up color mapping, showcasing the differences in methylation states among samples in the TCGA-CHOL dataset.</p>
<p>Additionally, using <code>colorRamp2()</code> ensures color comparability across multiple heatmaps. For example, in two heatmaps, the same color always corresponds to the same value:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(standardized_methylation_matrix, <span class="at">name =</span> <span class="st">"Standardized_Level1"</span>, <span class="at">col =</span> col_fun)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(standardized_methylation_matrix<span class="sc">/</span><span class="dv">2</span>, <span class="at">name =</span> <span class="st">"Standardized_Level2"</span>, <span class="at">col =</span> col_fun)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>p1 <span class="sc">+</span> p2 </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-colorRamp2complexheatmap" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-colorRamp2complexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-colorRamp2complexheatmap-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-colorRamp2complexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;22: Using colorRamp2 complexheatmap
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap uses <code>colorRamp2()</code> to enable color comparisons on the same scale, showcasing the differences in methylation states among samples in the TCGA-CHOL dataset.</p>
<p>If the matrix is continuous, you can simply provide a color vector, and the colors will be linearly interpolated. However, this method is not robust to outliers, as color mapping is determined by the minimum and maximum values in the matrix. Extreme outliers can cause less noticeable color variations within the normal range.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(standardized_methylation_matrix, <span class="at">name =</span> <span class="st">"Standardized_Level"</span>, <span class="at">col =</span> <span class="fu">rev</span>(<span class="fu">cm.colors</span>(<span class="dv">10</span>)), <span class="at">column_title =</span> <span class="st">"color vector for continuous matrix"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-ColorVectorcomplexheatmap" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ColorVectorcomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-ColorVectorcomplexheatmap-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ColorVectorcomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;23: Using Color Vector complexheatmap
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap demonstrates color mapping through linear interpolation, showing the differences in methylation states among samples in the TCGA-CHOL dataset.</p>
<p>You can also visualize NA values using the <code>na_col</code> parameter to specify the color for NA:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>standardized_methylation_matrix_na <span class="ot">&lt;-</span> standardized_methylation_matrix</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>na_index <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="cn">TRUE</span>, <span class="cn">FALSE</span>), <span class="fu">nrow</span>(standardized_methylation_matrix)<span class="sc">*</span><span class="fu">ncol</span>(standardized_methylation_matrix), <span class="at">replace =</span> <span class="cn">TRUE</span>, <span class="at">prob =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">9</span>))</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>standardized_methylation_matrix_na[na_index] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(standardized_methylation_matrix_na, <span class="at">name =</span> <span class="st">"matrix with na"</span>, <span class="at">na_col =</span> <span class="st">"grey"</span>, <span class="at">column_title =</span> <span class="st">"matrix with na"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-nacolcomplexheatmap" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-nacolcomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-nacolcomplexheatmap-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-nacolcomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;24: Using <code>na_col</code> Parameter complexheatmap
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap uses the <code>na_col</code> parameter to set the color of NA values to gray, showcasing the differences in methylation states among samples in the TCGA-CHOL dataset.</p>
<p>By default, linear interpolation of colors follows the LAB color space. Other color spaces, such as RGB, XYZ, and sRGB, are also available.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>f1 <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">seq</span>(<span class="fu">min</span>(standardized_methylation_matrix), <span class="fu">max</span>(standardized_methylation_matrix), <span class="at">length =</span> <span class="dv">3</span>), <span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"#EEEEEE"</span>, <span class="st">"red"</span>))</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>f2 <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">seq</span>(<span class="fu">min</span>(standardized_methylation_matrix), <span class="fu">max</span>(standardized_methylation_matrix), <span class="at">length =</span> <span class="dv">3</span>), <span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"#EEEEEE"</span>, <span class="st">"red"</span>), <span class="at">space =</span> <span class="st">"RGB"</span>)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(standardized_methylation_matrix, <span class="at">name =</span> <span class="st">"Standardized_Level1"</span>, <span class="at">col =</span> f1, <span class="at">column_title =</span> <span class="st">"color space in LAB"</span>)</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(standardized_methylation_matrix, <span class="at">name =</span> <span class="st">"Standardized_Level2"</span>, <span class="at">col =</span> f2, <span class="at">column_title =</span> <span class="st">"color space in RGB"</span>)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>p1 <span class="sc">+</span> p2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-DifferentLinearInterpolationcomplexheatmap" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-DifferentLinearInterpolationcomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-DifferentLinearInterpolationcomplexheatmap-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-DifferentLinearInterpolationcomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;25: Different Linear Interpolation complexheatmap
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap illustrates the differences in methylation states among samples in the TCGA-CHOL dataset across various color spaces.</p>
</section>
<section id="borders-and-gridlines" class="level4">
<h4 class="anchored" data-anchor-id="borders-and-gridlines">5.3 Borders and Gridlines</h4>
<p>The outermost border of the heatmap is set using the border parameter, while gridlines are configured with the <code>rect_gp</code> parameter.</p>
<p>The <code>border</code> parameter can be a logical or a color value. The <code>rect_gp</code> parameter requires a <code>grid::gpar</code> object.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(standardized_methylation_matrix, <span class="at">name =</span> <span class="st">"Standardized_Level"</span>, <span class="at">border =</span> <span class="st">"black"</span>, <span class="at">col =</span> col_fun, <span class="at">rect_gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">"white"</span>, <span class="at">lwd =</span> <span class="dv">2</span>) )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-BordersandGridlinescomplexheatmap" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-BordersandGridlinescomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-BordersandGridlinescomplexheatmap-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-BordersandGridlinescomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;26: Borders and Gridlines complexheatmap
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap modifies the borders and gridlines to show the differences in methylation states among samples in the TCGA-CHOL dataset.</p>
</section>
<section id="clustering" class="level4">
<h4 class="anchored" data-anchor-id="clustering">5.4 Clustering</h4>
<p>Adjusting the <code>cluster_rows</code> and <code>cluster_columns</code> parameters modifies the clustering of the heatmap.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(standardized_methylation_matrix, <span class="at">cluster_rows =</span> F, <span class="at">cluster_columns =</span> F)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>p1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-Clusteringcomplexheatmap" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-Clusteringcomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-Clusteringcomplexheatmap-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-Clusteringcomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;27: Clustering complexheatmap
</figcaption>
</figure>
</div>
</div>
</div>
<p>This heatmap disables clustering, highlighting the differences in methylation states among samples in the TCGA-CHOL dataset.</p>
</section>
<section id="annotation-bars-and-multiple-heatmaps" class="level4">
<h4 class="anchored" data-anchor-id="annotation-bars-and-multiple-heatmaps">5.5 Annotation Bars and Multiple Heatmaps</h4>
<p>Annotation bars are an important component of heatmaps. The <code>ComplexHeatmap</code> package provides flexible settings for annotation bars, which can be placed on any of the four sides of the heatmap (top, bottom, left, or right) using <code>top_annotation</code>, <code>bottom_annotation</code>, <code>left_annotation</code>, and <code>right_annotation</code>. All annotation bars are created using the <code>HeatmapAnnotation()</code> function (alternatively, the <code>rowAnnotation()</code> function can be used as a variant by specifying <code>HeatmapAnnotation(..., which = "row")</code>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">rowAnnotation</span>(</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Average Methylation"</span> <span class="ot">=</span> <span class="fu">anno_barplot</span>(</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rowMeans</span>(standardized_methylation_matrix), </span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">"blue"</span>), </span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="fu">unit</span>(<span class="dv">4</span>, <span class="st">"cm"</span>),</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis_param =</span> <span class="fu">list</span>(<span class="at">at =</span> <span class="fu">pretty</span>(<span class="fu">rowMeans</span>(standardized_methylation_matrix)), </span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">labels =</span> <span class="fu">format</span>(<span class="fu">pretty</span>(<span class="fu">rowMeans</span>(standardized_methylation_matrix)), <span class="at">nsmall =</span> <span class="dv">1</span>, <span class="at">digits =</span> <span class="dv">1</span>))</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(standardized_methylation_matrix, </span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">name =</span> <span class="st">"Standardized Methylation Levels"</span>, </span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>               <span class="at">col =</span> f2, </span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">row_km =</span> <span class="dv">2</span>,</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>               <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>))  <span class="co"># Based on row clustering</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>p1 <span class="sc">+</span> p2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-MultipleHeatmapscomplexheatmap" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-MultipleHeatmapscomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="Heatmap_files/figure-html/fig-MultipleHeatmapscomplexheatmap-1.png" class="img-fluid figure-img" style="width:95.0%">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-MultipleHeatmapscomplexheatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;28: Multiple Heatmaps complexheatmap
</figcaption>
</figure>
</div>
</div>
</div>
<p>This bar chart serves as an annotation bar for the heatmap, displaying the average methylation value for each site.</p>
<p>One of the standout features of the <code>ComplexHeatmap</code> package is its ability to plot multiple heatmaps simultaneously. The example below shows horizontally arranged heatmaps; vertically arranged heatmaps follow a similar principle. Horizontally aligned heatmaps require the same rows.</p>
<p>The combination of the heatmap and bar chart displays differences in methylation states among samples in the TCGA-CHOL dataset and the average methylation value for each site.</p>
</section>
</section>
</section>
<section id="applications" class="level2">
<h2 class="anchored" data-anchor-id="applications">Applications</h2>
<section id="basic-heatmap-4" class="level3">
<h3 class="anchored" data-anchor-id="basic-heatmap-4">1. Basic Heatmap</h3>
<div id="fig-HeatApplications1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-HeatApplications1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/Heatmap_app1.jpg" class="img-fluid figure-img" style="width:70.0%" alt="HeatApp1"></p>
<figcaption>Yang, M., Liu, B. et al.(2013)</figcaption>
</figure>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-HeatApplications1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;29: Applications of Basic Heatmap
</figcaption>
</figure>
</div>
<p>Figure 2: Heatmaps of TEC matching from 52 healthy individuals. (a) Raw heatmap. (b) Binary heatmap. [1]</p>
</section>
<section id="heatmaps-with-color-annotations-1" class="level3">
<h3 class="anchored" data-anchor-id="heatmaps-with-color-annotations-1">2. Heatmaps with Color Annotations</h3>
<div id="fig-HeatApplications2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-HeatApplications2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/Heatmap_app2.jpg" class="img-fluid figure-img" style="width:40.0%" alt="HeatApp2"></p>
<figcaption>Key, M. (2012)</figcaption>
</figure>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-HeatApplications2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;30: Applications of Heatmaps with Color Annotations
</figcaption>
</figure>
</div>
<p>Figure 6: Grouping by labels. All heatmap functions support row and column annotations, but implementations vary. Heatmap (a) and heatmap.2 (b) functions support displaying only a single color bar. Heatmap.plus (c) supports displaying a matrix of colored bars. Heatmap_plus (d) can display data frames for binary variables. Although heatmap.2 and heatmap.plus can generate rectangular images, heatmap and heatmap_plus output square images. [2]</p>
</section>
<section id="comparative-heatmaps" class="level3">
<h3 class="anchored" data-anchor-id="comparative-heatmaps">3. Comparative Heatmaps</h3>
<div id="fig-HeatApplications3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-HeatApplications3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/Heatmap_app3.jpg" class="img-fluid figure-img" style="width:40.0%" alt="HeatApp3"></p>
<figcaption>Hoggard, T. et al.(2016)</figcaption>
</figure>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-HeatApplications3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;31: Applications of Comparative Heatmaps
</figcaption>
</figure>
</div>
<p>Figure 6: Comparative heatmaps for miniARS experiments to resolve ambiguities in partitioning and replication functions associated with miniARS chromosomal fragments. Fragments depleted in the Cen-miniARS or Acen-miniARS experiments (e.g., miniARS317 fragments) compared to fragments enriched in these experiments. (B) Shared fragments in the miniARS datasets located in the given competition fitness deciles (i.e., fragments present in both Acen-miniARS and Cen-miniARS libraries) are represented in this heatmap. The heatmap is divided into 100 sections, each representing a combination of competition fitness deciles in the Acen-miniARS and Cen-miniARS experiments, as shown: Bottom-left: Fragments with low competition fitness in both Acen- and Cen-miniARS, indicating weak replication and partitioning abilities. (Partitioning “−”; Replication “−”). Contains 288 fragments representing 62 origins. Bottom-right: Fragments with high competition fitness in Acen-miniARS but low in Cen-miniARS, indicating strong partitioning but weak replication abilities (Partitioning “+”; Replication “−”). Contains 103 fragments representing 32 origins. Top-left: Fragments with low competition fitness in Acen-miniARS but high in Cen-miniARS, indicating weak partitioning but strong replication abilities (Partitioning “−”; Replication “+”). Contains 105 fragments representing 33 origins. (C) Combination analysis of the fragments from (B), aiming to define regions critical for maximizing partitioning abilities (1st and 4th pairs of heatmaps) versus replication abilities (2nd and 3rd pairs of heatmaps). (D) Similar to (B), but heatmaps include only miniARS fragments indicating ARS regions.<br>
Abbreviations: ARS: Autonomous Replicating Sequence, FKH: Forkhead, ORC: Origin Recognition Complex, RAP1: Rap1 protein binding site [3]</p>
</section>
</section>
<section id="reference" class="level2">
<h2 class="anchored" data-anchor-id="reference">Reference</h2>
<ol type="1">
<li>Yang, M., Liu, B., Zhao, M., Li, F., Wang, G., &amp; Zhou, F. (2013). Normalizing electrocardiograms of both healthy persons and cardiovascular disease patients for biometric authentication. <em>PLoS One, 8</em>(8), e71523. <a href="https://doi.org/10.1371/journal.pone.0071523" class="uri">https://doi.org/10.1371/journal.pone.0071523</a><br>
</li>
<li>Key, M. (2012). A tutorial in displaying mass spectrometry-based proteomic data using heat maps. <em>BMC Bioinformatics, 13</em>(Suppl 16), S10. <a href="https://doi.org/10.1186/1471-2105-13-S16-S10" class="uri">https://doi.org/10.1186/1471-2105-13-S16-S10</a><br>
</li>
<li>Hoggard, T., Liachko, I., Burt, C., Meikle, T., Jiang, K., Craciun, G., Dunham, M. J., &amp; Fox, C. A. (2016). High throughput analyses of budding yeast ARSs reveal new DNA elements capable of conferring centromere-independent plasmid propagation. <em>G3 (Bethesda), 6</em>(4), 993–1012. <a href="https://doi.org/10.1534/g3.116.027904" class="uri">https://doi.org/10.1534/g3.116.027904</a><br>
</li>
<li>Wickham, H. (2016). <em>ggplot2: Elegant graphics for data analysis</em>. Springer. <a href="https://r-graph-gallery.com/ggplot2-package.html" class="uri">https://r-graph-gallery.com/ggplot2-package.html</a><br>
</li>
<li>Rudis, B. (2020). <em>hrbrthemes: Additional themes and theme components for ‘ggplot2’</em>. <a href="https://cran.r-project.org/package=hrbrthemes" class="uri">https://cran.r-project.org/package=hrbrthemes</a><br>
</li>
<li>Wickham, H., François, R., Henry, L., &amp; Müller, K. (2021). <em>dplyr: A grammar of data manipulation</em>. <a href="https://cran.r-project.org/package=dplyr" class="uri">https://cran.r-project.org/package=dplyr</a><br>
</li>
<li>Wickham, H., &amp; Henry, L. (2021). <em>tidyr: Tidy messy data</em>. <a href="https://cran.r-project.org/package=tidyr" class="uri">https://cran.r-project.org/package=tidyr</a><br>
</li>
<li>Garnier, S. (2018). <em>viridis: Default color maps from ‘matplotlib’</em>. <a href="https://cran.r-project.org/package=viridis" class="uri">https://cran.r-project.org/package=viridis</a><br>
</li>
<li>Henry, L. (2022). <em>tibble: Simple data frames</em>. <a href="https://cran.r-project.org/package=tibble" class="uri">https://cran.r-project.org/package=tibble</a><br>
</li>
<li>Urbanek, S. (2021). <em>htmlwidgets: HTML widgets for R</em>. <a href="https://cran.r-project.org/package=htmlwidgets" class="uri">https://cran.r-project.org/package=htmlwidgets</a><br>
</li>
<li>Neuwirth, E. (2014). <em>RColorBrewer: ColorBrewer palettes</em>. <a href="https://cran.r-project.org/package=RColorBrewer" class="uri">https://cran.r-project.org/package=RColorBrewer</a><br>
</li>
<li>Winston, L., &amp; Hsu, T. (2021). <em>plotly: Create interactive web graphics via ‘plotly.js’</em>. <a href="https://cran.r-project.org/package=plotly" class="uri">https://cran.r-project.org/package=plotly</a><br>
</li>
<li><em>D3heatmap: Interactive heatmaps using ‘d3.js’</em>. <a href="https://cran.r-project.org/package=d3heatmap" class="uri">https://cran.r-project.org/package=d3heatmap</a><br>
</li>
<li>Gang, C. (2020). <em>heatmaply: Interactive heat maps for R</em>. <a href="https://cran.r-project.org/package=heatmaply" class="uri">https://cran.r-project.org/package=heatmaply</a><br>
</li>
<li>Becker, R. A., Wilks, A. R., &amp; Brownrigg, R. (2021). <em>lattice: The lattice package for R</em>. <a href="https://cran.r-project.org/package=lattice" class="uri">https://cran.r-project.org/package=lattice</a><br>
</li>
<li>Auguie, B. (2017). gridExtra: Miscellaneous Functions for “Grid” Graphics. <a href="https://cran.r-project.org/package=gridExtra" class="uri">https://cran.r-project.org/package=gridExtra</a></li>
</ol>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/openbiox\.github\.io\/Bizard\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<input type="hidden" id="giscus-base-theme" value="preferred_color_scheme">
<input type="hidden" id="giscus-alt-theme" value="preferred_color_scheme">
<script>
  function loadGiscus() {
    // Function to get the theme based on body class
    const getTheme = () => {
      let baseTheme = document.getElementById('giscus-base-theme').value;
      let altTheme = document.getElementById('giscus-alt-theme').value;
      return document.body.classList.contains('quarto-dark') ? altTheme : baseTheme;
    };
    const script = document.createElement("script");
    script.src = "https://giscus.app/client.js";
    script.async = true;
    script.dataset.repo = "openbiox/Bizard";
    script.dataset.repoId = "R_kgDONWVY7Q";
    script.dataset.category = "Announcements";
    script.dataset.categoryId = "DIC_kwDONWVY7c4ClcJT";
    script.dataset.mapping = "pathname";
    script.dataset.reactionsEnabled = "1";
    script.dataset.emitMetadata = "0";
    script.dataset.inputPosition = "top";
    script.dataset.theme = getTheme();
    script.dataset.lang = "en";
    script.crossOrigin = "anonymous";
    // Append the script to the desired div instead of at the end of the body
    document.getElementById("quarto-content").appendChild(script);
  }
  loadGiscus();
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../Correlation/Scatter.html" class="pagination-link" aria-label="Scatter Plot">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Scatter Plot</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../Correlation/Correlogram.html" class="pagination-link" aria-label="Correlogram">
        <span class="nav-page-text">Correlogram</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>© CC-By Peng Lab, 2024-2025. All rights reserved.</p>
</div>   
    <div class="nav-footer-center">
<p><a href="https://clustrmaps.com/site/1c4xe" title="Visit tracker"><img src="//clustrmaps.com/map_v2.png?cl=ffffff&amp;w=a&amp;t=n&amp;d=VK6_Uhjas4vA0CDps3EFeB0Fotb8hU50SYT4Fcq5nUI&amp;co=91c5eb&amp;ct=000000"></a></p>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/openbiox/Bizard/blob/main/Correlation/Heatmap.qmd" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/openbiox/Bizard/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This page is built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




</body></html>
